---
title: "Alpine snow algae blooms vary with elevation"
author:
  - name: Casey B. Engstrom
    institute: SFU
  - name: Kurt M. Yakimovich
    institute: SFU
  - name: Lynne M. Quarmby
    institute: SFU
    email: quarmby@sfu.ca
institute:
  - SFU: Dept. Molecular Biology and Biochemistry, Simon Fraser University, 8888 University Drive, Burnaby, BC, V5A 1S6, Canada
output: 
  bookdown::word_document2:
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=templates/scholarly-metadata.lua'
      - '--lua-filter=templates/author-info-blocks.lua'
    reference_docx: templates/word-template.docx
  # pdf_document:
  #   fig_caption: yes
  #   number_sections: yes
csl: the-isme-journal.csl
bibliography: refs.bib

---

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### *Keywords*: snow algae, microbiome, amplicon, *rbcL*, 18S, alpine, metabarcoding

### *Run title*: Alpine snow algae blooms


<!-- search for ??? before send out -->

##### pagebreak

# Abstract {-}

Snow algae blooms cover vast areas of summer snowfields worldwide, and can reduce albedo and increase snow melt. Despite their global prevalence, little is known about the algae species that comprise snow algae blooms. We used 18S and *rbcL* metabarcoding and light microscopy to characterise species composition of 33 snow algae samples from alpine and subalpine habitats in the Coast Range of British Columbia, Canada. We found blooms were dominated by the snow algae genera *Sanguina*, *Chloromonas*, and *Chlainomonas*. Above treeline blooms contained *Sanguina*, *Raphidonema*, and one OTU of *Chloromonas*, distinct from species assemblages below treeline predominated by *Chloromonas krienitzii*. Previously unrecognized as an important component of the snow algae microbiome, *Chlainomonas* were abundant in samples across all elevations. Our *rbcL* dataset detected diversity that was obscured by the low taxonomic resolution of traditional 18S metabarcoding, including OTUs that could represent novel species of snow algae. These three cross-referenced datasets show alpine snow algae blooms are more diverse than previously thought, and are highly variable from site to site along an elevational gradient.

<!-- need to add the few meters part -->


##### pagebreak

# Introduction {-}
<!-- global, vast blooms. albedo->snowmelt. what species? -->
Each summer in polar and alpine snowfields worldwide, vast areas of snow surface are colored red by snow algae blooms. Such blooms are reported from every continent [@segawa_bipolar_2018;@duval_snow_1999;@marchant_snow_1982;@yoshimura_community_1997;@vimercati_nieves_2019] and Arctic sea ice [@gradinger_snow_1996]. They can be extensive: in Alaska, snow algae spectral signatures were detected in one third of a 1,900 km^2^ icefield [@ganey_role_2017]. In recent years snow algae have received attention for their role in reducing snow surface albedo, which could substantially impact snow melt rates [@lutz_biogeography_2016;@ganey_role_2017]. Thus, snow algae could impact the timing of spring melt and runoff, depleting summer water supplies held in mountain snowpacks and reducing glacier mass balance. Despite their global ubiquity, we are only beginning to identify the species that cause snow algae blooms.

<!-- why microscopy not reliable -->
Much of the previous work on snow algae relied on cell morphology for identification, but microscopy-based identification of snow algae can be unreliable. Different species of snow algae can look nearly identical: recent phylogenetic work found that four distinct field samples dominated by similar-looking cells previously referred to as *Chloromonas cf. nivalis* are actually genetically distinct clades [@matsuzaki_taxonomic_2019]. Furthermore, the same species can look completely different in different environmental conditions or life stages. For example, cultured *Chloromonas krienitzii* cells are small green biflagellates, but field samples cells of this species are larger orange spheres with thick walls and short spines [@matsuzaki_revisiting_2015]. In the freshwater green algae *Haematococcus pluvialis* red secondary pigments are produced in response to high light [@shah_astaxanthin-producing_2016], but genetic comparison of red and green snow found they contained different species [@terashima_microbial_2017;@lutz_integrated_2015].

<!-- species from snow algae blooms, what is currently known -->
A diversity of algae have been reported from snow algae blooms, with many new species only having been described recently. Genera from the class Chlorophyceae are predominant in many blooms, including *Sanguina*, *Chloromonas*, and *Chlainomonas*. The taxonomy of *Sanguina* was recently established, and currently contains just two species; however, many sequences from red snow samples form an unnamed sister clade to *Sanguina* [@prochazkova_sanguina_2019]. The same study found *Sanguina nivaloides* predominated in red snowfields worldwide, while *S. aurantia* occurred in just three orange snow samples in Svalbard. The genus *Chloromonas* is speciose, containing twelve species that were isolated from snow [@matsuzaki_taxonomic_2019]. Different species of *Chloromonas* can form green, orange, or brownish-red blooms on the snow surface, and this genus has been reported from both polar and mid-latitude snowfields [@remias_ecology_2018;@remias_insights_2013;@prochazkova_chloromonas_2018]. Blooms of *Chlainomonas* have been found in red snow overlying alpine lakes in central Europe, western USA, and New Zealand [@novis_two_2008;@remias_ecophysiology_2016;@prochazkova_ecophysiological_2018]. This genus, which currently contains two named species, is closely related to *Chloromonas*. Non-Chlorophyceaen snow algae include Chrysophyceae found in Antarctica, the Alps, and Svalbard [@soto_microbial_2020;@remias_two_2019], and Trebouxiophyceae found in green snow in Greenland [@lutz_integrated_2015].

<!-- metabarcoding -->
In contrast to the high diversity of snow algae species observed in culture and Sanger sequencing, metabarcoding suggests that snow algae blooms are highly similar across continents, and are dominated by few taxa. 33 red snow communities across the Arctic were remarkably similar to each other: all were dominated by two operational taxonomic units (OTUs) of "uncultured Chlamydomonadaceae 1 and 2" with low relative abundance of *Chloromonas polyptera* and *Raphidonema nivale* [@lutz_biogeography_2016]. Similarly, another study using ITS2 metabarcoding found 24 red snow sites across the Arctic and Antarctic contained similar assemblages, dominated by "uncultured Chlamydomonadaceae A and B" with secondary abundance of *Raphidonema* and *Chloromonadinia* [@segawa_bipolar_2018]. Other 18S metabarcoding studies of snow algae were limited to genus or family level taxonomy [@hamilton_primary_2017]---because 18S is highly conserved, this marker generally cannot distinguish between closely related taxa.

<!-- our goal, what we did, what we found -->
Virtually nothing is known about the regional variation in species composition of snow algae blooms. Our goal was to identify the algae in blooms on different mountains and elevations in the Coast Range of British Columbia, Canada. We used 18S and *rbcL* metabarcoding to assess species composition---the latter targeting a hypervariable region of *rbcL* (coding for large subunit of rubisco), thus targeting only photosynthetic species with high taxonomic resolution. Additionally, we used light microscopy to describe the relative abundance of different morphologies in each field sample. By using these three cross-referenced metrics of relative abundance we were able to describe the algal species assemblage with high accuracy. Using *rbcL* as a marker revealed much greater diversity than would have been found using 18S alone. We did not find any differences in bloom composition between mountains, but there were distinct algae species assemblages at high versus low elevations.
<!-- stronger hook, predictions??? cliff hangers -->



# Materials and Methods {-}

<!-- refer to each supp at least once in txt -->
<!-- add cities, states -->

## Field sampling and microscopy {-}

We collected snow algae from sites throughout the Coast Range near Vancouver, British Columbia, Canada over the summer of 2018 (Supplementary Figure S1). To capture the extent of snow algae diversity we sampled from different elevations, dates, mountains, and micro-habitats within the snow. In total, we collected 309 colored snow samples from a range of 880 - 2150 m elevation, including samples from 13 different mountains collected on 33 different dates (Supplementary Table S1). Our samples trended upwards in elevation over the course of the summer as the snow melted earliest at the lowest elevations (Supplementary Figure S2). We first detected snow algae above 1500 m on June 20. We scooped samples from visibly colored snow into sterile 50 mL centrifuge tubes. To prevent melting en route to the lab we stored tubes in a bag of snow. We melted each sample at room temperature on the bench and removed a 1 mL aliquot for light microscopy. Immediately after, samples were stored at -20 &deg;C for up to eight months until DNA extraction. 

We used light microscopy to characterize the cell morphologies in each sample within 24 hours following collection. We fixed samples in 2% gluteraldehyde, and under 400x magnification observed the 100 cells closest to the center of the slide. We classified cell morphology based on similarity to published photographs of *Sanguina nivaloides* [@prochazkova_sanguina_2019], *Chloromonas cf. nivalis* [@prochazkova_chloromonas_2018], *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015], *Chloromonas krienitzii* [@matsuzaki_revisiting_2015], *Chlainomonas rubra* [@novis_two_2008], and for cells that did not fall into one of there categories as either "green cell" or "other". 

<!-- how do select field and micro photos fit into this? -->

## DNA extraction and amplicon library preparation {-}

<!-- This part is non reproducible, selecting based on date, etc -->

We selected 33 out of our 310 samples for *rbcL* and 18S Illumina amplicon sequencing. We chose these samples to represent the variation in date, elevation, geographic location, snow color, micro-habitat, and cell morphology. We freeze-dried these samples for up to 48 hours until until samples appeared completely desiccated. We mini-pestled between 5 to 20 mg of sample at room temperature to physically rupture cell walls before incubation in CTAB lysis buffer [@noauthor_ctab_2009]. We extracted DNA using chloroform:isoamyl alcohol [@cubero_dna_1999], and spin columns (Qiagen, Hilden) (detailed protocol in Supplementary File S1). As a negative control we processed a sterile distilled water sample alongside each batch.

<!-- arent we following some extraction protocol here? cant we cite, and then have a supplementary protocol with our variation??? -->

<!-- To extract DNA from the crushed cells we added 800 mL  , 1% B-mercaptoethanol, 5 $\mu$L each of proteinase K and RNAase A, and incubated these at 65 &deg;C for 30 minutes. We spun samples at 10,000 g for 3 minutes to pellet cell debris, then added 700 $\mu$L of supernatant to an equal volume of 24:1 chloroform:isoamyl alcohol. We inverted samples to mix and centrifuged again at 12,000 g for 10 minutes. We precipitated DNA by transferring the top layer to 700 $\mu$L ice-cold ethanol, which we gently mixed, and spun over Qiagen DNA columns for 30 s at 15,000 g. We washed columns twice with 70% ethanol, and finally dissolved our DNA by spinning with 50 $\mu$L of sterile TE buffer.  -->

We designed custom primers to target a hypervariable region of snow algae *rbcL* (coding for the large subunit of rubisco). *rbcL* is more differentiated between snow algae than 18S (Supplementary Figure S3), and contains similar number of reference sequences on GenBank. Our *rbcL* primers target a 400 bp section of snow algae *rbcL* based off the consensus of 20 GenBank snow algae sequences (Supplementary Table S2).  We designed our primers with the Eurofins primer design tool (https://www.eurofinsgenomics.eu/en/ecom/tools/pcr-primer-design/). *Sanguina* sequences were not included because they were not available at the time. To target 18S, we used the universal primers Euk1181 and Euk1624 targeting the V7-V8 hypervariable regions [@wang_optimal_2014]. Primer sequences are available in Supplementary Table S3. 

We constructed our 18S and *rbcL* amplicon libraries using a standard two-step PCR approach [@meyer_illumina_2010]. For detailed PCR conditions, see Supplementary File S2. After each PCR, we purified using Agencourt AMPure XP kit (Beckman Coulter, Brea, CA). We standardized sample DNA concentration using Qubit fluorometric quantification (Thermo Fisher, Waltham, MA). The pooled library was loaded and run on a MiSeq V3 kit (Illumina, San Diego, CA).


## Bioinformatic processing {-}

We demultiplexed reads using CUTADAPT [@martin_cutadapt_2011], and filtered and trimmed, removed errors, dereplicated, merged paired-end reads, and removed chimeras following the default pipeline of DADA2 [@callahan_dada2_2016]. We assigned taxonomy to amplicon sequence variants (ASVs) using IDTaxa [@murali_idtaxa_2018], only assigning names to ASVs that were assigned with high confidence (threshold set to 50). We assigned taxonomy using custom snow algae reference databases for both 18S and *rbcL*, in addition to re-assigning 18S with SILVA to classify non-snow algae [@quast_silva_2013]. Snow algae reference databases and the scripts to generate them are available on our github repository (see below).

<!-- should cite a github script here??? singletons, failed samples removed, remove sequences based on negative controls, adjust for 0's, sequencing depth, -->

<!-- make sure this is on GH: -->

<!-- We made our *rbcL* database by downloading 30,865 algae rbcL sequences from GenBank with the query “rbcl[gene] AND (algae OR chlorophyta OR trebouxiophyceae) NOT (18s OR ribosomal OR psaB OR atpB) AND 300:2000[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion”. We removed ambiguous snow algae annotations: because entries labelled as *Chloromonas cf. brevispina* and *Chloromonas cf. nivalis* are polyphyletic [@matsuzaki_taxonomic_2019] we re-labeled these as “unassigned *Chloromonas*”.  and then re-ran ASVs assigned to Chlorophyta on our custom 18S snow algae database based on the following GenBank query: "(18S OR ribosomal)[gene] AND (chloromonas OR chlainomonas OR sanguina OR raphidonema OR KMY-2018) NOT (rbcL OR psaB OR atpB) AND 300:3500[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion". -->

To generate OTUs we clustered ASVs using t-SNE [@maaten_visualizing_2008]. We compared t-SNE results with the perplexity parameter ranging from 1 to 100, and found comparable results from 10 to 50. We categorized OTUs by applying DBSCAN [@hahsler_dbscan_2019] to our t-SNE. We compared *rbcL* sample similarity using UniFrac with default settings [@lozupone_unifrac_2005]. To confirm that our clustering was not an artifact of t-SNE we created maximum likelihood phylogenetic trees of ASVs using IQTree [@nguyen_iq-tree:_2015] (Supplementary Figures S4-S5).

All raw sequence data are available under European Nucleotide Archive accession PRJEB34539. All scripts used in this study are available at https://github.com/cengstro/bc_snow_algae_amplicon.



# Results {-}


<!-- add read depth, etc? -->
<!-- make fig 5, ???, and make 18s otus -->

We observed morphologically distinct snow algae communities at different elevations (Figure 1). Red snow was prevalent in areas of high solar exposure above treeline, predominantly containing cells similar in appearance to *Sanguina nivaloides* [@prochazkova_sanguina_2019]. We occasionally observed green snow patches in alpine meadows receiving full sun, but more frequently observed green snow in well-shaded, forested sites. We found cells resembling *Chlainomonas rubra* [@novis_two_2008] at all elevations. Forested sites at low elevations often contained green or orange patches of snow, predominated by green or orange ovate cells with short spines resembling *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015].

Both 18S and *rbcL* amplicon libraries were dominated by ASVs that were assigned to Chlorophyta. Our 18S library contained 67 ASVs that were assigned to Chlorophyta and 7 ASVs to Ochrophyta, while our *rbcL* library contained 603 ASVs assigned to Chlorophyta and 41 to Trebouxiophyceae. Negative controls did not contain Qubit-detectable amounts of DNA. Our *rbcL* library contained higher ASV and OTU diversity than 18S: t-SNE clustering generated 9 *rbcL* OTUs, compared with just 4 18S OTUs (Supplementary Figure S6). The most frequent *rbcL* taxonomic assignments at genus level were *Chloromonas*, *Chlainomonas*, and *Sanguina*. Although most ASVs were not assigned to genus level, most ASVs clustered near one of these genera in NMDS ordination (Figure 2). Several *rbcL* OTUs were dominated by ASVs that were not assigned to genus level, including three OTUs closely related to *Chloromonas*, and one Chlamydomonadaceae OTU that was only assigned to the family level.

The composition of 18S and *rbcL* libraries varied with elevation (Figure 3; Figure 4). *Sanguina* was absent below 1500 m in both 18S and *rbcL*. High elevation samples contained an unannotated OTU of *Chloromonas* that was absent from low elevation sites (Figure 3, "Chloromonas C"). Additionally, the three highest elevation sites uniquely contained ASVs with a best BLAST match to *Raphidonema longiseta* (KM462868.1)---interestingly, all of these samples were taken from snow overlying glacier. The two high alpine green snow samples were dominated by two different OTUs of *Chloromonas*, which were also both found in sites below treeline. *Chlainomonas* was prevalent at all elevations, and *Chloromonas krienitzii* was predominant in our low elevation sites.

We observed snow algae blooms dominated by *Chloromonas krienitzii* transition from green to orange from May to June (Figure 5). At two sites we observed snow melt channels or runnels containing high concentrations of snow algae. In May and early June these sites contained green snow hidden 2 to 5 cm below the white snow surface, but upon subsequent visits the same location contained orange snow at the surface, whose morphologies resembled published images of the *Chloromonas krienitzii* lineage [@matsuzaki_revisiting_2015]. We compared 18S and *rbcL* metabarcode profiles of green subsurface and orange surface snow, and found both contained high relative abundance of *Chloromonas krienitzii*. Although we did not detect *Chlainomonas* in cell counts, the surface samples contained nearly double the relative abundance of *Chlainomonas* from the subsurface samples.


<!-- include faiths diversity in supps??? thinking no  -->

# Discussion {-}

Here we present multiple data sets demonstrating elevational patterns in alpine snow algae blooms. *Sanguina* was dominant in red snow above treeline, while green and orange blooms of *Chloromonas krienitzii* were dominant in mid-elevation snow runnels. We found unexpected diversity within *Chloromonas* and *Chlainomonas* *rbcL* that we did not detect using 18S, including many potentially novel OTUs. In contrast to previous studies [@novis_two_2008;@prochazkova_ecophysiological_2018], our work suggests that *Chlainomonas* is widespread and abundant in alpine snow algae blooms. 

Overall, our results show bloom composition can be highly variable on the same mountain, even in nearby patches. This is in contrast to previous papers that found snow algae assemblages were homogeneous across geographically distant sites [@lutz_biogeography_2016;@segawa_bipolar_2018]. This could be due to different sampling schemes: the aforementioned studies exclusively sampled high-latitude red snow, while our study included mid-latitude red, orange, and green snow from above and below treeline. Another recent study using metabarcoding found different snow algae communities in red and green snow algae blooms in Antarctica, including yellow blooms dominated by Chrysophyceae [@soto_microbial_2020]. Much of the diversity in our samples within *Chlainomonas* and *Chloromonas* was only detected with *rbcL* and not 18S, so the diversity within these genera may have been overlooked in previous metabarcoding studies.

Our *rbcL* data suggest *Chlainomonas* may be more common than previously thought. We could not differentiate between *Chlainomonas* and *Chloromonas* in our 18S data set, so this genus may have been overlooked in previous studies. *Chlainomonas* was previously thought to be restricted to waterlogged snow overlying mountain lakes [@novis_two_2008;@prochazkova_ecophysiological_2018], but we did not find any evidence that *Chlainomonas* was restricted in its habitat distribution. To the contrary, only one *Chlainomonas* dominant sample was located in waterlogged snow near the edge of a pool (sample S9); the other *Chlainomonas* dominant sites were not notably wetter than the surrounding snow, nor overlying frozen bodies of water. Interestingly, we found *Chlainomonas* was in higher relative abundance in our *rbcL* dataset than in cell counts (Figure 3)---possibly due to multiple chloroplasts per cell, which have been documented in this genus [@prochazkova_ecophysiological_2018].

Our findings highlight the remaining unexplored diversity in the snow algae microbiome. Many *rbcL* OTUs did not closely match any GenBank nucleotide archive sequences (Figure 2). "Chlamydomonadaceae E" highest BLAST percent identity was only 87%, and the top 10 BLAST matches contained 6 different algae genera---including two *Chloromonas* snow algae from Japan. This OTU did not correlate with any categories in our 18S or cell count data. One possible explanation is that it was lumped together with *Chloromonas* in 18S, and our crude quantification and categorization of cell counts did not detect this taxa. Other OTUs were also poorly matched to GenBank,  which could represent novel species of *Chloromonas* (Figure 2). The lack of clustering we observed in the remaining *Chloromonas* could be due to overlap in the levels of interspecific and intraspecific variation. *rbcL* diversity is likely higher within *Chloromonas* than in other algae genera: unlike *Sanguina* [@prochazkova_sanguina_2019], most species of *Chloromonas* lack a pyrenoid [@nozaki_differences_2002], which houses high concentrations of cross-linked rubisco. Likely related to its lack of pyrenoid, *Chloromonas* has many non-synonymous mutations in the region of *rbcL* the codes for binding rubisco together [@nozaki_differences_2002]. While *rbcL* is problematic in *Chloromonas* phylogeny [@nozaki_differences_2002], it nonetheless is highly differentiated and therefore an effective barcode for microalgae [@hall_assessment_2010].

<!-- *Sanguina* was found in the same samples, but relative abundance was generally higher in 18S and cell counts (Figure 3). Our *rbcL* primers were designed based on *Chloromonas* and *Chlainomonas* snow algae sequences (*Sanguina* sequence data was not available at the time), so primer bias could have resulted in over-representation of *Chloromonas* relative to *Sanguina*. -->

Variation in bloom composition could be due to a wide range of habitat features. *Sanguina* and *Chloromonas C* were limited to sites above 1500 m in full sunlight, but many low elevation sites also received full sunlight and did not contain these OTUs (Figure 3). Light intensity, snow moisture, or snow chemistry could all plausibly influence community composition. Intriguingly, we only observed *Raphidonema* at high-elevation glacier sites. In Svalbard, *Raphidonema nivale* abundance increased on glacier surface snow following wind storms, and the authors suggest that this is a soil algae that grows sub-optimally on the snow surface following wind deposition [@stibal_growth_2005]. Given the aerial dispersal capabilities of microalgae [@tesson_airborne_2016] and genetic overlap between distant snow algae populations [@segawa_bipolar_2018] it seems unlikely that geographic distance is a barrier to snow algae distribution. Underlying topography could potentially influence species distribution: two sites dominated by *Chloromonas krienitzii* were in runnels overlying ephemeral streams, which could influence nutrient availability where surface debris is deposited in the runnel. 

Over multiple visits we observed motile green *Chloromonas krienitzii* cells below the snow surface develop into an orange bloom at the surface (Figure 5). Previous work has shown that this species can appear as distinct green and orange morphologies [@matsuzaki_revisiting_2015], but this paper is the first to document a bloom dominated by this species change color in the field. The transition over weeks suggests that this process is mediated by seasonal changes. Secondary pigments likely protect snow algae from the damaging effects of intense solar irradiation at the snow surface [@bidigare_evidence_1993], which could be why we only observed green cells below the snow surface.

In conclusion, our study reveals substantial diversity within and between snow algae blooms. We found greater diversity using *rbcL* metabarcoding than 18S, which suggests snow algae diversity may have been overlooked in studies that used only 18S. Species were differentially distributed along an elevational gradient, with distinct blooms dominated by *Chloromonas*, *Sanguina*, or *Chlainomonas*. Future studies will examine other taxa present in snow algae blooms, and their interactions in these globally important microbiomes.




# Acknowledgments {-}

We thank Leah Tooman (Simon Fraser University) for assistance with sequencing, and Chris Rushton (Simon Fraser University) for assistance with bioinformatics. This project was funded with a Sector Innovation Grant from Genome BC (SIP016), and a NSERC Individual Discovery Grant, both awarded to LQ. 

# Author Contributions {-}

CE, LQ, and KY designed this study. Samples collected by CE with assistance from KY and LQ. KY and CE prepped samples for sequencing. All analyses were completed by CE and KY. All authors discussed the results and contributed to the final manuscript. CE wrote the manuscript with major input from LQ and KY.

# Additional information {-}

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.





##### pagebreak

# References {-}
<div id="refs"></div> 


##### pagebreak

# Figure legends {-}

<!-- formatting indentation??? -->

**Figure 1.** Representative photographs of snow algae. A. Red snow bloom above treeline at sample site G1.1. B. Microphotograph of red snow containing *Sanguina nivaloides* (Sn) and *Chlainomonas rubra* (Cr) cell morphologies. All scale bars 30 $\mu$m, all microphotos taken using 63x objectives with DIC. C. Snow runnels in a forest clearing containing snow algae (inset). D. Microphotograph of orange snow from the surface of runnel containing *Chloromonas krienitzii* (Ck) and *Chloromonas cf. nivalis* (Cn). E. Bronze coloured snow algae blooms below conifer canopy. F. Microphotograph of bronze snow containing Chloromonas cf. brevispina* (Cb) and *Chloromonas cf. nivalis* (Cn).

\setlength{\leftskip}{0pt}
**Figure 2.** Ordination plots of *rbcL* ASV similarity. A. Multidimensional scaling (MDS) plot. Taxonomy (assigned by IDTaxa) indicated by color, and size of point is proportional to cumulative relative abundance. Dotted ellipses show OTU clusters. Stress = 0.13. B. t-Distributed Stochastic Neighbor Embedding (t-SNE) dimensionality reduction of snow algae *rbcL* ASVs. Dotted lines indicate OTUs. Perplexity = 30.

**Figure 3.** Stacked bar plots of snow algae relative abundance. Each sample is arranged on the y axis in order of elevation, with each compositional bar plot representing the (left to right) cell morphology, 18S, and *rbcL* snow algae assemblages of the same sample. Colors below each bar plot indicate taxonomic categories. Morphospecies were identified by light microscopy, using similarity to published photographs as a guide. 18S OTUs are reference-based, while *rbcL* OTUs were clustered *de novo*, and named using GenBank reference data.

**Figure 4.** Non-metric multidimensional scaling (NMDS) showing *rbcL* UniFrac distances between samples. Each point represents a sample, which is labelled by sample ID and elevation (color).

**Figure 5.** ??? Need to make the figure first