---
title: "Alpine snow algae species composition along an elevational gradient"
author:
  - name: Casey B. Engstrom
    institute: SFU
  - name: Kurt M. Yakimovich
    institute: SFU
  - name: Lynne M. Quarmby
    institute: SFU
    email: quarmby@sfu.ca
institute:
  - SFU: Dept. Molecular Biology and Biochemistry, Simon Fraser University, 8888 University Drive, Burnaby, BC, V5A 1S6, Canada
output: 
  bookdown::word_document2:
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=templates/scholarly-metadata.lua'
      - '--lua-filter=templates/author-info-blocks.lua'
    reference_docx: templates/word-template.docx
  # pdf_document:
  #   fig_caption: yes
  #   number_sections: yes
csl: templates/the-isme-journal.csl
bibliography: refs.bib
---

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### *Keywords*: snow algae, microbiome, amplicon, *rbcL*, 18S, alpine, metabarcoding

### *Run title*: Alpine snow algae blooms

The authors declare no conflicts of interest.


<!-- search for ??? before send out -->

##### pagebreak

# Abstract {-}

Snow algae blooms cover vast areas of summer snowfields worldwide, and can reduce albedo and increase snow melt. Despite their global prevalence, little is known about the algae species that comprise snow algae blooms. We used 18S and *rbcL* metabarcoding and light microscopy to characterize species composition of 33 snow algae samples from alpine and subalpine habitats in the Coast Range of British Columbia, Canada. We found blooms were dominated by the snow algae genera *Sanguina*, *Chloromonas*, and *Chlainomonas*. This study is the first to thoroughly document regional variation between blooms, and we found that blooms were highly variable in species composition, with high elevation sites containing distinct species assemblages from forested sites. In contrast to previous work, the snow algae genus *Chlainomonas* was an abundant and widespread in snow algae blooms. Although we found few taxa using traditional 18S metabarcoding, *rbcL* revealed diversity with high taxonomic resolution, including OTUs that could represent novel species of snow algae. These three cross-referenced datasets show alpine snow algae blooms are more diverse than previously thought, and are highly variable from site to site along an elevational gradient.

<!-- need to add the few meters part -->


##### pagebreak

# Introduction {-}
<!-- global, vast blooms. albedo->snowmelt. what species? -->
Each summer, vast areas of snow surface are coloured red by snow algae blooms in polar and alpine snowfields worldwide. Red snowfields have been found on every continent [@segawa_bipolar_2018;@duval_snow_1999;@marchant_snow_1982;@yoshimura_community_1997;@vimercati_nieves_2019] and overlying Arctic sea ice [@gradinger_snow_1996]. The distribution and extent of snow algae blooms are poorly understood, but they can be quite extensive: in Alaska, remote sensing detected snow algae in one third of a 1,900 km^2^ icefield [@ganey_role_2017]. In recent years snow algae have received attention for their role in reducing snow surface albedo, which could substantially impact the rate of snow melt [@lutz_biogeography_2016;@ganey_role_2017]. Potentially, snow algae could substantially impact the timing of spring melt, which in turn could deplete summer water supplies in mountain snowpack and reduce glacier mass balance. Snow algae blooms have been recorded throughout history since the time of the ancient Romans [@mayhoff_naturalis_1906;@darwin_voyage_1959], yet despite their global prevalence we are only beginning to identify the species that cause snow algae blooms.

Microscopy reveals a diversity of cell morphologies in snow algae blooms, but different species can look nearly identical [@matsuzaki_taxonomic_2019], and the same species can look completely different under different environmental conditions. The snow algae *Chloromonas krienitzii* are small green biflagellates in culture, but in field samples cells of this species are nearly twice the diameter, orange-pigmented, with thick cell walls and short spines [@matsuzaki_revisiting_2015]. The environmental cues that trigger this transformation are not well understood, but in the freshwater green algae *Haematococcus pluvialis* red secondary pigments are produced in response to high light [@shah_astaxanthin-producing_2016]. Green snow algae blooms are less frequently reported, and microscopy-based studies have suggested that green snow develops into red snow [@muller_persistent_2001]. However, metabarcoding studies to date have found that green snow and red snow contain different species [@terashima_microbial_2017;@lutz_integrated_2015].

<!-- species from snow algae blooms, what is currently known -->
A diversity of algae have been reported from snow algae blooms, with many new species only having been described recently. Genera from the class Chlorophyceae are predominant in many blooms, including *Sanguina*, *Chloromonas*, and *Chlainomonas*. The taxonomy of *Sanguina* was recently established, and currently contains just two species; however, many sequences from red snow samples form an unnamed sister clade to *Sanguina* [@prochazkova_sanguina_2019]. The same study found *Sanguina nivaloides* predominated in red snowfields worldwide, while *S. aurantia* occurred in just three orange snow samples in Svalbard. The genus *Chloromonas* is speciose, containing twelve species that were isolated from snow [@matsuzaki_taxonomic_2019]. Different species of *Chloromonas* can form green, orange, or brownish-red blooms on the snow surface, and this genus has been reported from both polar and mid-latitude snowfields [@remias_ecology_2018;@remias_insights_2013;@prochazkova_chloromonas_2018]. Blooms of *Chlainomonas* have been found in red snow overlying alpine lakes in central Europe, western USA, and New Zealand [@novis_two_2008;@remias_ecophysiology_2016;@prochazkova_ecophysiological_2018]. This genus, which currently contains two named species, is closely related to *Chloromonas*. Non-Chlorophyceaen snow algae include Chrysophyceae found in Antarctica, the Alps, and Svalbard [@soto_microbial_2020;@remias_two_2019], and Trebouxiophyceae found in green snow in Greenland [@lutz_integrated_2015].

<!-- metabarcoding, red snow looks the same -->
Although there are many species of snow algae, only a few OTUs seem to dominate red snow algae blooms worldwide. 33 Arctic red snow samples were all dominated by "uncultured Chlamydomonadaceae 1 and 2" OTUs with low relative abundance of *Chloromonas polyptera* and *Raphidonema nivale* [@lutz_biogeography_2016]. Similarly, another study using metabarcoding found 24 red snow sites across the Arctic and Antarctic contained similar assemblages, dominated by "uncultured Chlamydomonadaceae A and B" with secondary abundance of *Raphidonema* and *Chloromonadinia* [@segawa_bipolar_2018]. Other 18S metabarcoding studies of snow algae were limited to genus or family level taxonomy [@hamilton_primary_2017]---because 18S is highly conserved, this marker cannot distinguish between closely related species or genera.

<!-- our goal, what we did, what we found -->
Virtually nothing is known about the regional variation in species composition of snow algae blooms, particularly in alpine regions. We set out to identify the algae in blooms on different mountains and elevations in the Coast Range of British Columbia, Canada. We assessed species composition using light microscopy and 18S and *rbcL* metabarcoding. *rbcL* codes for the large subunit of Rubisco, thus targeting only the photosynthetic component of the microbiome. Using these three cross-referenced metrics we describe the algal species assemblage in our study area with high accuracy. *rbcL* revealed much greater diversity than would have been found using 18S alone. We did not find any differences in bloom composition between mountains, but there were distinct algae species assemblages at different elevations.



# Materials and Methods {-}


## Field sampling and microscopy {-}

We collected 309 snow algae samples over the summer of 2018 from alpine and subalpine sites in the Coast Range near Vancouver, British Columbia, Canada (Supplementary Figure S1). We collected red, orange, and green snow samples from 13 different mountains from sites ranging from 880 to 2150 m above sea level (Supplementary Table S1). Over the summer we collected samples from progressively higher elevations as snow melted at lower elevations, and we did not detect snow algae at higher elevations until later in the summer. We scooped samples from visibly coloured snow into sterile 50 mL centrifuge tubes. We kept samples cold during transport back to the lab by packing tubes in snow. We melted each sample at room temperature on the bench and removed a 1 mL aliquot for light microscopy. Samples were then stored at -20 &deg;C for up to eight months until DNA extraction. 

We used light microscopy to characterize cell morphology within 24 hours after collection. We fixed samples in 2% gluteraldehyde, and under 400 x magnification observed the 100 cells closest to the center of the slide. We classified cell morphology based on similarity to published photographs of *Sanguina nivaloides* [@prochazkova_sanguina_2019], *Chloromonas cf. nivalis* [@prochazkova_chloromonas_2018], *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015], *Chloromonas krienitzii* [@matsuzaki_revisiting_2015], *Chlainomonas rubra* [@novis_two_2008], and for cells that did not fall into one of there categories as either "green cell" or "other". 

<!-- collected a ton of samples, characterized morphology -->

## DNA extraction and amplicon library preparation {-}

<!-- arising implicit hypothesis 1) different morph species, different genetic species 2) different elevation, different species (our methods not ideal for this, should have been randomized sample if we wanted to test for elevation) -->

We Illumina sequenced *rbcL* and 18S amplicons of 33 samples. We selected this subset for sequencing to compare the species composition of blooms containing distinct morphologies, and to compare the genetic composition of morphologically similar blooms from different mountains, elevations, and dates. We freeze-dried these samples for up to 48 hours until until samples appeared completely desiccated. We mini-pestled between 5 to 20 mg of sample at room temperature to physically rupture cell walls before incubation in CTAB lysis buffer [@noauthor_ctab_2009]. We extracted DNA in batches of 5-6 samples using chloroform:isoamyl alcohol [@cubero_dna_1999], and purified DNA using ethanol and spin columns (Qiagen, Hilden) (Supplementary Protocol S1). As a negative control against cross-contamination we processed a tube of sterile distilled water alongside each batch, and tested this for DNA with Qubit fluorometric quantification (Thermo Fisher, Waltham, MA).

In addition to 18S, we designed custom snow algae *rbcL* primers. We chose this marker because *rbcL* is more differentiated between snow algae than 18S (Supplementary Figure S2), and there is snow algae *rbcL* reference data on GenBank. Our primers were designed to target a 400 bp section of snow algae *rbcL*, based off the consensus of 20 GenBank snow algae sequences (Supplementary Figure S2). *Sanguina* sequences were not included because they were not available at the time. We designed our primers with the Eurofins primer design tool (https://www.eurofinsgenomics.eu/en/ecom/tools/pcr-primer-design/). To target 18S, we used the universal primers Euk1181 and Euk1624 targeting the V7-V8 hypervariable regions [@wang_optimal_2014]. Primer sequences are available in Supplementary Table S2. 

We constructed our 18S and *rbcL* amplicon libraries using a standard two-step PCR approach [@meyer_illumina_2010] (Supplementary Protocol S2). After each PCR, we purified using Agencourt AMPure XP kit (Beckman Coulter, Brea, CA). We standardized sample DNA concentration with Qubit, pooled our library, and sequenced this on a MiSeq using the V3 kit following manufacturer instructions (Illumina, San Diego, CA).


## Bioinformatic processing {-}

We demultiplexed reads with CUTADAPT [@martin_cutadapt_2011], and filtered, trimmed, denoised, dereplicated, merged paired-end reads, and removed chimeras following the default pipeline of DADA2 [@callahan_dada2_2016]. We assigned taxonomy to amplicon sequence variants (ASVs) using IDTaxa [@murali_idtaxa_2018] using a high confidence threshold parameter (t=50). Because snow algae are not well represented on curated databases such as SILVA, we made custom databases using GenBank snow algae sequences (scripts and databases available on github). To ensure that we were not overlooking non-snow algae taxa we also ran our 18S library against SILVA [@quast_silva_2013]. 

<!-- should cite a github script here??? singletons, failed samples removed, remove sequences based on negative controls, adjust for 0's, sequencing depth, -->

<!-- ??? make sure this is on GH: -->

<!-- We made our *rbcL* database by downloading 30,865 algae rbcL sequences from GenBank with the query “rbcl[gene] AND (algae OR chlorophyta OR trebouxiophyceae) NOT (18s OR ribosomal OR psaB OR atpB) AND 300:2000[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion”. We removed ambiguous snow algae annotations: because entries labelled as *Chloromonas cf. brevispina* and *Chloromonas cf. nivalis* are polyphyletic [@matsuzaki_taxonomic_2019] we re-labeled these as “unassigned *Chloromonas*”.  and then re-ran ASVs assigned to Chlorophyta on our custom 18S snow algae database based on the following GenBank query: "(18S OR ribosomal)[gene] AND (chloromonas OR chlainomonas OR sanguina OR raphidonema OR KMY-2018) NOT (rbcL OR psaB OR atpB) AND 300:3500[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion". -->

To assign OTUs, we clustered ASVs using t-SNE [@maaten_visualizing_2008] and DBSCAN [@hahsler_dbscan_2019]. To ensure that our t-SNE was not an artifact of parameter selection, we ran t-SNE with different values of "perplexity", and found the same clusters were generated from 10 to 50. To cross-validate this unconventional OTU clustering method, we overlaid these OTU clusters on phylogenetic trees using IQTree [@nguyen_iq-tree:_2015] (Supplementary Figure S3).

<!-- mention swarm? would need to show this dint work on tree as supp, no time -->

# Results {-}

<!-- add read depth, etc? -->
<!-- i think the cell counts needs to go into the main text -->

We observed morphologically distinct snow algae blooms at different elevations (Figure 1; Supplementary Figure S4). Red snow was prevalent in areas of high solar exposure above treeline, predominantly containing cells similar in appearance to *Sanguina nivaloides* [@prochazkova_sanguina_2019]. We found cells resembling *Chlainomonas rubra* [@novis_two_2008] across all elevations. Green snow patches were infrequent in alpine meadows receiving full sun, and more common in shady forested sites. Forested sites often contained green or orange patches of snow, many of these sites contained green or orange ovate cells with short spines resembling *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015].

Both 18S and *rbcL* amplicon libraries were dominated reads assigned to Chlorophyta (Figure 2). Our 18S library assigned with SILVA contained 50 Chlorophyceae, 11 Trebouxiophyceae, and 7 Chrysophyceae ASVs, while our *rbcL* library contained 603 Chlorophyta and 41 Trebouxiophyceae ASVs. Both ASV and OTU diversity was higher in *rbcL* than 18S: nine *rbcL* OTUs, compared to just three in 18S (Supplementary Figure S5). The most frequent *rbcL* genera were *Chloromonas*, *Chlainomonas*, and *Sanguina*. Although the majority of ASVs were not assigned to genus level, most ASVs were genetically similar to one of these three genera, with one exception. Sequences in the OTU E were not assigned to genus level; the ten best BLAST matches included six different genera within Chlamydomonadaceae (86-87% sequence match), two of which were *Chloromonas* snow algae (LC012735). Many ASVs were similar to *Chloromonas*, including two distinct OTU clusters D and F which did not match any known species within this genus.  

18S and *rbcL* taxonomic composition varied with elevation. Low elevation samples clustered together in *rbcL* UniFrac [@lozupone_unifrac_2005], forming a cluster that was distinct from most high elevation samples (Figure 3). We found the highest diversity in samples that were collected latest in the season, although we did not find a statistically significant trend between Shannon diversity and date there was a weak correlation between Faith's PD and date (Pearson's *r*=0.36, p=0.04) (Supplementary Figure S6). In both 18S and *rbcL*, *Sanguina* predominated many sites above 1500 m, but was absent below this elevation (Figure 4). Additionally, high elevation samples contained one OTU of *Chloromonas* that was absent from low elevation sites (Figure 3, OTU F). The three highest elevation sites uniquely contained ASVs with a best BLAST match to *Raphidonema longiseta* (KM462868.1)---interestingly, all of these samples were taken from snow overlying glacier. The two high alpine green snow samples were dominated by two different OTUs of *Chloromonas*, which were also both found in sites below treeline. *Chlainomonas* was prevalent at all elevations, and *Chloromonas krienitzii* was predominant in our low elevation sites.

A green to orange transition occurred from May to June at two sites dominated by *Chloromonas krienitzii*. On initial visits to a site in May we observed green snow hidden 2 to 5 cm below the snow surface (sample S2.2), but on subsequent visits in June (samples S4.6, S6.1, S9.2, S11.2) the same runnel was orange on the surface overlying the green snow. Microscopy of orange snow revealed orange spherical cells with short spines resembling *Chloromonas krienitzii* [@matsuzaki_revisiting_2015], while green snow contained motile green flagellates. *rbcL* comparison of green and orange snow showed both were dominated by reads assigned to *Chloromonas krienitzii*, with the orange snow containing higher abundance of *Chlainomonas* (Supplementary Figure S7).


# Discussion {-}

Here we present multiple data sets demonstrating elevational patterns in alpine snow algae blooms. *Sanguina* was dominant in red snow above treeline, while green and orange blooms of *Chloromonas krienitzii* were dominant in runnels at lower elevations. We found unexpected diversity within *Chloromonas* and *Chlainomonas* *rbcL* that we did not detect using 18S, including many potentially novel OTUs. Previously only reported from red snow overlying lakes [@prochazkova_ecophysiological_2018], *Chlainomonas* was widespread and abundant in alpine snow algae blooms. 

Our results show bloom composition can be highly variable on the same mountain, even in nearby patches. In contrast, previous studies found red snow algae assemblages were homogeneous across geographically distant polar sites [@lutz_biogeography_2016;@segawa_bipolar_2018]. This discrepancy could be because our study included green and orange snow samples, or it could be that mid-latitude mountains have higher snow algae diversity than polar sites. We noted most snow algae blooms above treeline were dominated by *Sanguina*; by only collecting a single sample as representative of a region could have meant that the previous studies could have overlooked less common snow algae communities. We detected more OTUs with *rbcL* than 18S, so previous 18S metabarcoding could have overlooked snow algae diversity at lower taxonomic levels.

Our *rbcL* data suggest *Chlainomonas* may be more widely distributed than previously thought. *Chlainomonas* was previously thought to be restricted to waterlogged snow overlying mountain lakes [@novis_two_2008;@prochazkova_ecophysiological_2018], but we did not find any evidence that *Chlainomonas* was restricted in its habitat distribution. To the contrary, only one *Chlainomonas* dominant sample was located in waterlogged snow near the edge of a pool (sample S9), while the other *Chlainomonas* dominant sites were not notably wetter than the surrounding snow, nor were they located over frozen lakes. We could not differentiate between *Chlainomonas* and *Chloromonas* in our 18S data set, so this genus may have remained undetected in previous 18S metabarcoding. Interestingly, *Chlainomonas* was in higher relative abundance in our *rbcL* dataset than in cell counts (Figure 3). One possible explanation is high *rbcL* copy number: the large cells of *Chlainomonas rubra* have multiple parietal chloroplasts per cell [@prochazkova_ecophysiological_2018], and *rbcL* is located in the plastid genome.

Our findings highlight the remaining unexplored diversity in the snow algae microbiome. Several *rbcL* OTUs did not closely match any GenBank nucleotide archive sequences, such as the OTU "Chlamydomonadaceae E" (Figure 2). This OTU was distinct in our *rbcL* data, but it did not correlate with any categories in 18S or cell count data. One possible explanation is that it was lumped together with *Chloromonas* in 18S, and our crude quantification and categorization of cell counts did not detect this taxa. OTUs D and F were related to *Chloromonas*, but contained few ASVs that were assigned to genus (Figure 2). Possibly these OTUs represent novel species of *Chloromonas*. However, there were no discernible clusters within the OTU "Other *Chloromonas*", perhaps due to overlap in the levels of interspecific and intraspecific variation. *rbcL* diversity is likely higher within *Chloromonas* than in other algae genera: unlike *Sanguina* [@prochazkova_sanguina_2019], most species of *Chloromonas* lack a pyrenoid [@nozaki_differences_2002], which houses high concentrations of cross-linked Rubisco. Lacking pyrenoid, *Chloromonas* have many non-synonymous mutations in the region of *rbcL* the codes for binding Rubisco together [@nozaki_differences_2002]. While *rbcL* is a poor indicator of *Chloromonas* phylogeny [@nozaki_differences_2002], it nonetheless is highly differentiated and therefore an effective barcode for microalgae [@hall_assessment_2010].

Variation in bloom composition could be due to a wide range of habitat features. *Sanguina* and *Chloromonas* C were limited to sites above 1500 m in full sunlight, but many low elevation sites also received full sunlight and did not contain these OTUs (Figure 3). Light intensity, snow moisture, or snow chemistry could all plausibly influence community composition. Intriguingly, we only observed *Raphidonema* at high-elevation glacier sites. In Svalbard, *Raphidonema nivale* abundance increased on glacier surface snow following wind storms, and the authors concluded that *Raphidonema* was a soil algae that opportunistically colonizes snow following wind deposition, but being sub-optimally adapted for snow such blooms are short lived [@stibal_growth_2005]. Given the aerial dispersal capabilities of microalgae [@tesson_airborne_2016] and genetic overlap between distant snow algae populations [@segawa_bipolar_2018] it seems unlikely that geographic distance is a barrier to snow algae distribution. Underlying topography could potentially influence species distribution: two sites dominated by *Chloromonas krienitzii* were in runnels overlying ephemeral streams, which could influence nutrient availability where surface debris is deposited in the runnel. 

<!-- As i presented it above, looks like Chlainomonas is causing the bloom. Need to explain the nuance -->
Previous work has shown that *Chloromonas krienitzii* undergoes distinct green and orange morphologies [@matsuzaki_revisiting_2015], but this paper is the first to document this transition in the field. The transition over weeks suggests that this process is mediated by seasonal changes. Secondary pigments likely protect snow algae from the damaging effects of intense solar irradiation at the snow surface [@bidigare_evidence_1993], which could be why green cells were most concentrated a few centimeters below the snow surface. *Chlainomonas* was more abundant in surface samples as measured by *rbcL*, but we microscopy suggested that *Chlainomonas* was only present in low relative abundance. 

In conclusion, our study reveals substantial diversity within and between snow algae blooms. We found greater diversity using *rbcL* metabarcoding than 18S, which suggests snow algae diversity may be overlooked in studies using only 18S. Species were differentially distributed along an elevational gradient, with distinct blooms dominated by *Chloromonas*, *Sanguina*, or *Chlainomonas*. Future studies will examine other taxa present in snow algae blooms, and their interactions in these globally important microbiomes.


# Data availability {-}

All raw sequence data are available under European Nucleotide Archive accession PRJEB34539. All scripts used in this study are available at https://github.com/cengstro/bc_snow_algae_amplicon. Supplementary materials are available at https://www.nature.com/ismej/.

# Acknowledgments {-}

We thank Leah Tooman (Simon Fraser University) for assistance with sequencing, and Chris Rushton (Simon Fraser University) for assistance with bioinformatics. This project was funded with a Sector Innovation Grant from Genome BC (SIP016), and a NSERC Individual Discovery Grant, both awarded to LQ. 

# Author contributions {-}

CE, LQ, and KY designed this study. Samples collected by CE with assistance from KY and LQ. KY and CE prepped samples for sequencing. All analyses were completed by CE and KY. All authors discussed the results and contributed to the final manuscript. CE wrote the manuscript with major input from LQ and KY.

# Conflict of interest {-}

The authors declare this research was conducted in the absence of any conflict of interest.


##### pagebreak

# References {-}
<div id="refs"></div> 


##### pagebreak

# Figure legends {-}

**Figure 1.** Representative photographs of snow algae. A. Red snow bloom above treeline at sample site G1.1. B. Microphotograph of red snow containing *Sanguina nivaloides* (Sn) and *Chlainomonas rubra* (Cr) cell morphologies. All scale bars 30 $\mu$m, all microphotos taken using 63x objectives with DIC. C. Snow runnels in a forest clearing containing snow algae (inset). D. Microphotograph of orange snow from the surface of runnel containing *Chloromonas krienitzii* (Ck) and *Chloromonas cf. nivalis* (Cn). E. Bronze coloured snow algae blooms below conifer canopy. F. Microphotograph of bronze snow containing Chloromonas cf. brevispina* (Cb) and *Chloromonas cf. nivalis* (Cn).

**Figure 2.** Ordination plots of *rbcL* ASV similarity. A. Multidimensional scaling (MDS) plot. Taxonomy (assigned by IDTaxa) indicated by colour, and size of point is proportional to cumulative relative abundance. Dotted ellipses show OTU clusters. Stress = 0.13. B. t-Distributed Stochastic Neighbor Embedding (t-SNE) dimensionality reduction of snow algae *rbcL* ASVs. Dotted lines indicate OTUs. Perplexity = 30.

**Figure 3.** Stacked bar plots of snow algae relative abundance. Each sample is arranged on the y axis in order of elevation, with each compositional bar plot representing the (left to right) cell morphology, 18S, and *rbcL* snow algae assemblages of the same sample. colours below each bar plot indicate taxonomic categories. Morphospecies were identified by light microscopy, using similarity to published photographs as a guide. 18S OTUs are reference-based, while *rbcL* OTUs were clustered *de novo*, and named using GenBank reference data.

**Figure 4.** Non-metric multidimensional scaling (NMDS) showing *rbcL* UniFrac distances between samples. Each point represents a sample, which is labelled by sample ID and elevation (colour).

