---
title: "Diverse snow algae blooms in the alpine and subalpine zones of British Columbia"
author:
  - name: Casey B. Engstrom
    institute: SFU
  - name: Kurt M. Yakimovich
    institute: SFU
  - name: Lynne M. Quarmby
    institute: SFU
    email: quarmby@sfu.ca
institute:
  - SFU: Dept. Molecular Biology and Biochemistry, Simon Fraser University, 8888 University Drive, Burnaby, BC, V5A 1S6, Canada
output: 
  bookdown::word_document2:
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=templates/scholarly-metadata.lua'
      - '--lua-filter=templates/author-info-blocks.lua'
    reference_docx: templates/word-template.docx
  # pdf_document: 
  #   fig_caption: yes
  #   number_sections: yes
csl: frontiers.csl
bibliography: refs.bib
abstract: |
  Snow algae blooms cover vast areas of summer snowfields worldwide, and can reduce albedo and increase snow melt. Despite their global prevalence, little is known about the species that comprise snow algae blooms. We used 18S and *rbcL* high-throughput amplicon sequencing (metabarcoding) to survey 33 snow algae samples from alpine and subalpine habitats in the Coast Range of British Columbia, Canada. Species composition varied from bloom to bloom: *Sanguina* predominated above treeline, while *Chloromonas* were prevalent at lower elevations. *Chlainomonas* were abundant in samples across all elevations. The highest diversity was contained within the genus *Chloromonas*, which included operational taxonomic units (OTUs) that likely represent novel species of snow algae. ???

---

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### *Keywords*: snow algae, microbiome, amplicon, *rbcL*, 18S, alpine, Illumina, British Columbia

### *Run title*: Alpine snow algae


<!-- search for ??? before send out -->



##### pagebreak

# Introduction {-}
<!-- global, vast blooms. albedo->snowmelt. what species? -->
Each summer in polar and alpine snowfields worldwide, vast areas of snow surface are colored red by snow algae blooms. Such blooms are reported from every continent [@segawa_bipolar_2018;@duval_snow_1999;@marchant_snow_1982;@yoshimura_community_1997;@vimercati_nieves_2019] and Arctic sea ice [@gradinger_snow_1996]. They can be extensive: in Alaska, snow algae spectral signatures were detected in one third of a 1,900 km^2^ icefield [@ganey_role_2017]. In recent years snow algae have received attention for their role in reducing snow surface albedo, which could substantially impact snow melt rates [@lutz_biogeography_2016;@ganey_role_2017]. Thus, snow algae could impact the timing of spring melt and runoff, reducing water supplies in mountain snowpacks and glacier mass balance. Despite their global ubiquity, we are only beginning to identify the species that cause snow algae blooms.

<!-- why microscopy not reliable -->
Much of the previous work on snow algae relied on cell morphology for identification, but microscopic identification of snow algae can be unreliable. The same species can look completely different with different environmental conditions of life stages. For example, cultured *Chloromonas krienitzii* cells are small green biflagellates, but field samples cells of this species are larger orange spheres with thick walls and short spines [@matsuzaki_revisiting_2015]. In the freshwater green algae *Haematococcus pluvialis* red pigment is produced in response to stress conditions [@shah_astaxanthin_2016], and @muller_persistant_2001 hypothesized that red Arctic snowfields develops from green snow, but thus far genetic studies have failed to support this hypothesis [@terashima_microbial_2017;@lutz_integrated_2015]. Different species of snow algae can look nearly identical: recent phylogenetic work found that four distinct field samples dominated by similar-looking cells previously referred to as *Chloromonas cf. nivalis* are actually genetically distinct clades [@matsuzaki_taxonomic_2019].

<!-- species from snow algae blooms, what is currently known -->
A diversity of algae have been reported from snow algae blooms. Chlorophyceae are predominant in many blooms, including the genera *Sanguina*, *Chloromonas*, and *Chlainomonas*. Sanger sequencing of red snow samples has shown that *Sanguina nivaloides* is predominant in red snowfields worldwide [@prochazkova_sanguina_2019]. The taxonomy of this genus is only recently established; while only one other species is ascribed to this genus, many sequences from red snow samples form an unnamed sister clade to *Sanguina*. The genus *Chloromonas* is speciose, with twelve species isolated from snow [@matsuzaki_taxonomic_2019], and different species can form green, orange, or brownish-red blooms on the snow surface [@remias_ecology_2018;@remias_insights_2013;@prochazkova_chloromonas_2018]. Blooms of *Chlainomonas* have been found in red snow overlying alpine lakes in central Europe, western USA, and New Zealand [@novis_two_2008;@remias_ecophysiology_2016;@prochazkova_ecophysiological_2018]. Two species are currently assigned to the genus *Chlainomonas*, but the taxonomy is not well-established. Non-Chlorophyceaen snow algae include Chrysophyceae found in Antarctica, the Alps, and Svalbard [@soto_microbial_2020;@remias_two_2019], and Trebouxiophyceae found in green snow [@lutz_integrated_2015].

<!-- metabarcoding -->
While many species of snow algae have been isolated or sequenced from snow, much less is known about the species composition of snow algae blooms: which species predominate, which species co-occur, and how species are distributed across the landscape. Previous studies using 18S amplicon high-throughput sequencing (metabarcoding) suggest that blooms are highly similar across continents. 33 red snow communities across the Arctic were remarkably similar to one another: all were dominated by an "uncultured Chlamydomonadaceae 1 and 2" with low relative abundance of *Chloromonas polyptera* and *Raphidonema nivale* [@lutz_biogeography_2016]. Similarly, another study using ITS2 metabarcoding found 24 red snow sites across the Arctic and Antarctic contained similar assemblages, dominated by "uncultured Chlamydomonadaceae A and B" with secondary abundance of *Raphidonema* and *Chloromonadinia* [@segawa_bipolar_2018]. Other metabarcoding studies were limited to genus or family level taxonomic resolution [@hamilton_primary_2017]. As 18S is highly conserved across eukaryotes, sequences can be too similar to resolve between closely related taxa.

<!-- our goal, what we did, what we found -->
Virtually nothing is known about the regional variation in species composition of snow algae blooms. Our goal was to identify the algae in blooms on different mountains and elevations in the Coast Range of British Columbia, Canada. We used 18S and *rbcL* metabarcoding to assess species composition---the latter targeting a hypervariable region of *rbcL* (coding for large subunit of rubisco), thus targeting only photosynthetic species with high taxonomic resolution. Additionally, we used light microscopy to describe the relative abundance of different morphologies in each field sample. By using these three cross-referenced metrics of relative abundance we were able to describe the algal species assemblage with high accuracy. We discovered a much greater diversity of snow algae than expected. Although we did not find any differences in bloom composition between mountains, there were distinct blooms at high versus low elevation.


# Methods {-}

## Field sampling and microscopy {-}

We collected snow algae from sites throughout the Coast Range near Vancouver, British Columbia, Canada over the summer of 2018. To capture the extent of snow algae diversity in our local mountains we collected as many samples as possible from different elevations, dates, mountains, and micro-habitats within the snow (S2). In total we collected 310 colored snow samples from 13 different mountains on 33 different dates from elevations between 880 m and 2150 m above sea level. In early season we sampled at lower elevations, moving uphill as the snow algae bloom progressed upwards in elevation. 

We scooped samples from visibly colored snow into 50 mL centrifuge tubes using sterile technique. To prevent melting en route to the lab we stored tubes in a bag of snow. Back in the lab, we melted each sample at room temperature on the bench and removed a 1 mL aliquot for light microscopy. Immediately after, samples were stored at -20 &deg;C for up to eight months until DNA extraction. 

We used light microscopy to characterize the cell morphologies in each sample within 24 hours following collection, with samples maintained on ice. We prepared a slide of cells fixed in 2% gluteraldehyde, and counted up to 100 cells at 400x. We classified each cell based on similarity to published photographs, as either *Sanguina nivaloides* [@prochazkova_sanguina_2019], *Chloromonas cf. nivalis* [@prochazkova_chloromonas_2018], *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015], *Chloromonas krienitzii* [@matsuzaki_revisiting_2015], *Chlainomonas rubra* [@novis_two_2008], and for cells that did not fall into one of there categories as either "green cell" or "other". 


## DNA extraction and amplicon library preparation {-}

We selected 33 out of 310 samples for rbcL and 18s high-throughput amplicon sequencing. We chose these samples to represent the variation in date, elevation, geographic location, snow color, micro-habitat, and cell morphology. We freeze-dried these samples for up to 48 hours until until samples appeared completely desiccated. We mini-pestled between 5 to 20 mg of sample at room temperature to physically rupture cell walls before incubation in lysis buffer.

To extract DNA from the crushed cells we added 800 mL 1x CTAB extraction buffer [@noauthor_ctab_2009], 1% B-mercaptoethanol, 5 $\mu$L each of proteinase K and RNAase A, and incubated these at 65 &deg;C for 30 minutes. We spun samples at 10,000 g for 3 minutes to pellet cell debris, then added 700 $\mu$L of supernatant to an equal volume of 24:1 chloroform:isoamyl alcohol. We inverted samples to mix and centrifuged again at 12,000 g for 10 minutes. We precipitated DNA by transferring the top layer to 700 $\mu$L ice-cold ethanol, which we gently mixed, and spun over Qiagen DNA columns for 30 s at 15,000 g. We washed columns twice with 70% ethanol, and finally dissolved our DNA by spinning with 50 $\mu$L of sterile TE buffer. As a negative control we processed a sterile distilled water sample alongside each batch, treating it exactly the same as the other tubes; all negative controls did not contain Qubit-detectable levels of DNA.
	
We designed custom primers to target a hypervariable region of *rbcL* of snow algae, along with established universal 18S V7-V8 primers Euk1181 and Euk1624 [@wang_optimal_2014]. We designed *rbcL* primers using the Eurofins primer design tool [@eurofins_pcr_nodate] to target a 400 bp section of *rbcL* based off of 20 snow algae GenBank sequences (Supplemental table 3)---*Sanguina* sequences were not included because they were not available at the time of this study. The primer sequences can be seen in Supplemental table 4. 

We constructed our amplicon library using a standard two-step PCR approach [@meyer_illumina_2010]. In the first PCR we amplified template DNA using our primers attached to a universal adapter, and in the second PCR we re-amplified that product to attach a 6 bp index to the universal adapter at the 3’ end. The first PCR total volume was 25 $\mu$L, consisting of 1 $\mu$L template, 12.5 $\mu$L Q5 high-fidelity 2X MM (NEB), 1.25 $\mu$L each of forward and reverse primer, and 9 $\mu$L of ddH2O. The second PCR was the same except we reduced our reaction volume to 20 $\mu$L by using only 5 $\mu$L of ddH2O. The cycling conditions were the same for both primer pairs for the first PCR, with an initial denaturation at 98 &deg;C for 30 s, followed by 30 cycles of 98 &deg;C for 5 s, 58 &deg;C for 10 s, and 72 &deg;C for 25 s, with a final extension at 72 &deg;C for 2 minutes. For the second indexing PCR we started with an initial denaturation at 98 &deg;C for 30 s, then 10 cycles of 98 &deg;C for 10 s, 65 &deg;C for 30 s, and 72 &deg;C for 30 s with a final denaturation of 72 &deg;C for 5 min. After each PCR, we purified using Agencourt AMPure XP kit (Beckman Coulter). We quantified final DNA concentration with Qubit (Thermo Fisher), and standardized sample concentration for pooling. The pooled library was then loaded and run on an Illumina MiSeq V3 kit.


## Bioinformatic processing {-}

Reads were demultiplexing using CUTADAPT [@martin_cutadapt_2011]. Samples were filtered and trimmed, errors removed, dereplicated, pair end reads merged, and chimeras removed following the default pipeline of DADA2 [@callahan_dada2_2017].  <!-- singletons, failed samples removed, remove sequences based on negative controls, adjust for 0's, sequencing depth, -->

We assigned taxonomy for both *rbcL* and 18S amplicon sequence variants (ASVs) using IDTaxa with threshold set to 50 [@murali_idtaxa_2018]. We made custom reference databases for each to include snow algae GenBank sequences. We made our rbcL database by downloading 30,865 algae rbcL sequences from GenBank with the query “rbcl[gene] AND (algae OR chlorophyta OR trebouxiophyceae) NOT (18s OR ribosomal OR psaB OR atpB) AND 300:2000[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion”. We removed ambiguous snow algae annotations: because entries labelled as *Chloromonas cf. brevispina* and *Chloromonas cf. nivalis* are polyphyletic [@matsuzaki_taxonomic_2019] we re-labeled these as “unassigned *Chloromonas*”. To assign 18S reads we initially ran our ASVs against SILVA [@quast_silva_2013], and then re-ran ASVs assigned to Chlorophyta on our custom 18S snow algae database based on the following GenBank query: "(18S OR ribosomal)[gene] AND (chloromonas OR chlainomonas OR sanguina OR raphidonema OR KMY-2018) NOT (rbcL OR psaB OR atpB) AND 300:3500[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion".

As most operational taxonomic unit (OTU) clustering algorithms are optimized for 18S data, we visualized *rbcL* ASV clustering using t-SNE [@maaten_visualizing_2008] (Fig. 2b). We tested this with perplexity values ranging from 1 to 100, and found comparable results with perplexity ranging from 10 to 50. We compared sample similarity using UniFrac with default settings [@lozupone_unifrac_2005]. 

For software information, see Supplementary Table 9. All raw fastq files are freely available on the European Nucleotide Archive under the project accession PRJEB34539. All scripts are available at https://github.com/cengstro/bc_snow_algae_amplicon.



# Results {-}

We collected 309 snow algae samples from 13 mountains throughout the summer of 2018. At our low elevation sites (850 - 1500 m) we detected snow algae as early as May 18, but these sites were also the earliest to melt out. We did not detect snow algae at high elevation sites (1500 - 2200 m) until June 20, where both the snow and snow algae blooms persisted throughout the entire summer.

Using light microscopy we identified morphologically distinct snow algae in different habitats (Fig. 1, S4). Red snow was prevalent in areas of high solar exposure; above treeline sites were often dominated by red cells similar in appearance to *Sanguina nivaloides* [@prochazkova_sanguina_2019] (Fig. 1a,b). We found cells resembling *Chlainomonas rubra* [@novis_two_2008] at all elevations, in highest relative abundance in areas receiving direct sunlight. We occasionally observed green snow patches in alpine meadows receiving full sun, but more frequently observed green snow in well-shaded, forested sites. At four low-angle sites in clearings near treeline we observed snow melt channels or runnels containing high concentrations of snow algae (Fig. 1c). In May and early June these sites contained green snow hidden 2 to 5 cm below the white snow surface, but upon subsequent visits the same location contained orange snow at the surface, whose morphologies resembled published images of the *Chloromonas krienitzii* lineage [@matsuzaki_revisiting_2015] (Fig. 1d). Forested sites at low elevations often contained green or orange patches of snow, containing green or orange ovate cells with short spines resembling *Chloromonas cf. brevispina* [@matsuzaki_revisiting_2015] (Fig. 1e,f), although we also occasionally observed red snow at forested sites containing morphologies similar to *Sanguina nivaloides*.

Of the 309 samples studied by light microscopy, we selected 33 snow algae samples for 18S and *rbcL* amplicon Illumina sequencing. Both libraries were dominated by amplicon sequence variants (ASVs) that were assigned to Chlorophyta. Our 18S library contained 67 ASVs that were assigned to Chlorophyta and 7 ASVs to Ochrophyta, while our *rbcL* library contained 642 ASVs, of which 603 were assigned to Chlorophyta, the remaining 41 were Trebouxiophyceae. The top genera assigned by the taxonomy assignment algorithm IDTaxa [@murali_idtaxa_2018] were *Chloromonas*, *Chlainomonas*, and *Sanguina*; although most ASVs were not assigned to genus level by this algorithm, the majority of ASVs clustered near one of these genera in NMDS ordination (Fig. 2a). We used t-SNE to generate operational taxonomic units (OTUs), and found that many OTUs that did not correspond to any known taxa on GenBank (Fig. 2b). These included three OTUs that were closely related to *Chloromonas*, and one Chlamydomonadaceae OTU that was only assigned to the family level, meaning that it did not match any known algae genus on GenBank.

Above and below treeline samples contained some unique taxa, but most taxa did not exhibit a clear elevational trend (Fig. 3). *Sanguina* was the dominant taxa in most samples from above treeline, and was absent in sequences libraries from the samples below 1500 m. Additionally, these high elevation samples uniquely contained an unannotated OTU of *Chloromonas* (Fig. 3, "Chloromonas C"). The three highest elevation sites uniquely contained ASVs with a best BLAST match to *Raphidonema longiseta* (KM462868.1)---all three samples were taken from snow overlying glacier . Green snow from high alpine sites contained *Chloromonas* OTUs that were also found in sites below treeline. Low elevation sites contained *Chloromonas krienitzii*, but many sites were dominated by taxa that were found at all elevations, such as *Chlainomonas*. 

Cell counts, 18S, and *rbcL* datasets were generally consistent, but there were some interesting discrepancies. *Chlainomonas* was more prevalent in our *rbcL* library than in cell counts, while *Sanguina* was underrepresented in *rbcL* compared to 18S and cell counts. We did not detect *Chlainomonas* with 18S sequencing. 

At runnel sites on Hollyburn and Seymour we observed snow algae blooms that changed from green to orange over the course of the summer (Fig. 1). We sequenced both green subsurface and orange surface snow at three different sites, and found both were dominated by *Chloromonas krienitzii*, although surface samples also contained *Chlainomonas* (Supplementary figure ???).


# Discussion {-}

We found that alpine snow algae bloom composition varied from site to site, with some elevational trends. *Sanguina* was dominant above treeline, while *Chloromonas krienitzii* was dominant in mid-elevation runnels, which initially appeared as green motile cells below the snow surface but developed into orange blooms at the snow surface. We found an unexpected diversity within *Chloromonas*, including many OTUs not represented in GenBank. Previous studies suggested that *Chlainomonas* is restricted to waterlogged snow overlying mountain lakes [@novis_two_2008;@prochazkova_ecophysiological_2018] but our results suggest that this genus can be widespread and abundant in many alpine red snow sites. While one *Chlainomonas*-dominant sample was located in waterlogged snow near the edge of a pool (sample S9), the other *Chlainomonas*-dominant sites were not notably wetter than the surrounding snow, nor overlying ice.

Our findings highlight the remaining unexplored diversity in the snow algae microbiome. *rbcL* revealed greater diversity than 18S, and many *rbcL* OTUs did not closely match any annotated GenBank sequences (Fig. 2). Likely, the *Chloromonas* *rbcL* OTUs were lumped together in the 18S dataset due to high similarity. We found higher *rbcL* diversity in *Chloromonas* than *Sanguina*, which could be because *Chloromonas* lack a pyrenoid, and consequently has many non-synonymous mutations in the region of *rbcL* the codes for binding these proteins together to form a pyrenoid [@nozaki_differences_2002]. One *rbcL* OTU, "Chlamydomonadaceae E", was only assigned to family level, which could represent a novel genus. However, this OTU did not correspond with any features in our 18S or cell count data. It seems unlikely that Chlamydomonadaceae E would have been lumped in with *Chloromonas* in 18S, given how distinct their *rbcL* sequences were from *Chloromonas*. Primers bias could explain this discrepancy: perhaps our *rbcL* primers over-represented this OTU compared to 18S.

Although 18S, *rbcL*, and cell counts were generally consistent, we observed some interesting discrepancies (Fig. 3). We did not observe *Chlainomonas* in our 18S dataset, despite the fact that there were two representative sequences from this genus in our reference database (MF803743.1, MF803745.1). Likely, *Chlainomonas* 18S in our target region (V7-V8) is not sufficiently differentiated from *Chloromonas* to be distinguishable. Many samples contained high relative abundance of *Chlainomonas* *rbcL*, yet these samples contained lower concentration of *Chlainomonas* cell morphologies. Large cells such as *Chlainomonas* often have high rRNA gene copy number [@fu_single_2017], which could account for this discrepancy. *Sanguina* was found in the same samples, but relative abundance was generally higher in 18S and cell counts. Our *rbcL* primers were designed based on *Chloromonas* and *Chlainomonas* snow algae sequences (*Sanguina* sequence data was not available at the time), so primer bias could have resulted in over-representation of *Chloromonas* relative to *Sanguina*.

Variation in bloom composition could be due to a wide range of habitat features. *Sanguina* and *Chloromonas C* were limited to sites above 1500 m in full sunlight, but many low elevation sites also received full sunlight and did not contain these OTUs (Fig. 3). Light intensity, snow moisture, or snow chemistry could all plausibly influence community composition. Intriguingly, we only observed *Raphidonema* at high-elevation glacier sites. In Svalbard *Raphidonema nivale* abundance increased on glacier surface snow following windstorms, and the authors suggest that this is a soil algae that grows sub-optimally on the snow surface following wind deposition [@stibal_growth_2005]. Underlying topography could also potentially influence species distribution: two sites dominated by *Chloromonas krienitzii* were in runnels overlying ephemeral streams, which could influence nutrient availability where surface debris is deposited in the snow runnel. Given the aerial dispersal capabilities of microalgae [@tesson_airborne_2016] and genetic overlap between distant snow algae populations [@segawa_bipolar_2018] it seems unlikely that geographic distance is a barrier for snow algae distribution.

Over multiple visits to sites S and H we observed green motile *Chloromonas krienitzii* cells below the snow surface develop into an orange bloom at the surface (Supplementary figure ???). Previous work has shown that this species can appear as distinct green and orange morphologies [@matsuzaki_revisiting_2015], but our observations are the first to our knowledge to describe this transition occurring in the field. The change over weeks suggests that this process is mediated by seasonal changes. Secondary pigments likely protect snow algae from the damaging effects of intense solar irradiation at the snow surface [@bidigare_evidence_1993], which could be why we only observed green cells below the snow surface.

In conclusion, our study reveals substantial diversity within and between snow algae blooms. Species were differentially distributed along an elevational gradient, with distinct blooms dominated by *Chloromonas*, *Sanguina*, or *Chlainomonas*. Future studies will examine other taxa present in snow algae blooms, and their interactions in these globally important microbiomes.




# Acknowledgments {-}

We thank Leah Tooman (Simon Fraser University) for assistance with sequencing, and Chris Rushton (Simon Fraser University) for assistance with bioinformatics. This project was funded with a Sector Innovation Grant from Genome BC (SIP016), and a NSERC Individual Discovery Grant, both awarded to LQ. 

# Author Contributions {-}

CE, LQ, and KY designed this study. Samples collected by CE with assistance from KY and LQ. KY and CE prepped samples for sequencing. All analyses were completed by CE and KY. All authors discussed the results and contributed to the final manuscript. CE wrote the manuscript with major input from LQ and KY.

# Additional information {-}

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.



l

##### pagebreak

# References {-}
<div id="refs"></div> 


##### pagebreak

# Figure legends {-}

<!-- formatting indentation? -->

**Figure 1.** Representative photographs of snow algae. A. Red snow bloom above treeline at sample site G1.1. B. Microphotograph of red snow containing *Sanguina nivaloides* (Sn) and *Chlainomonas rubra* (Cr) cell morphologies. All scale bars 30 $\mu$m, all microphotos taken using 63x objectives with DIC. C. Snow runnels in a forest clearing containing snow algae (inset). D. Microphotograph of orange snow from the surface of runnel containing *Chloromonas krienitzii* (Ck) and *Chloromonas cf. nivalis* (Cn). E. Bronze coloured snow algae blooms below conifer canopy. F. Microphotograph of bronze snow containing Chloromonas cf. brevispina* (Cb) and *Chloromonas cf. nivalis* (Cn).


**Figure 2.** Ordination plots of *rbcL* amplicon sequence variant (ASV) similarity. A. Multidimensional scaling (MDS) plot. Taxonomy (assigned by IDTaxa) indicated by color, and size of point is proportional to cumulative relative abundance. Dotted ellipses show OTU clusters. Stress = 0.13. B. t-Distributed Stochastic Neighbor Embedding (t-SNE) dimensionality reduction plot of snow algae *rbcL* ASVs. Perplexity = 30.

**Figure 3.** Stacked bar plots of snow algae relative abundance. Each sample is arranged on the y axis in order of elevation, with each compositional bar plot representing the (left to right) cell morphology, 18S, and *rbcL* snow algae assemblages of the same sample. Colors below each bar plot indicate taxonomic categories. Morphospecies were identified by light microscopy, using similarity to published photographs as a guide. 18S operational taxonomic units (OTUs) are reference-based, while *rbcL* OTUs were clustered *de novo*, and named using GenBank reference data.

**Figure 4.** Non-metric multidimensional scaling (NMDS) showing *rbcL* UniFrac distances between samples. Each point represents a sample, which is labelled by sample ID and elevation (color).