---
title: "Overlapping snow algae communities in the alpine and subalpine zones of British Columbia"
# who grows in the snow? multi-hued snow algae communities
# different coloured alpine and subalpine snows harbour constrasting snow algae communities
# contrasting snow algae communities in di
# not just dirty snow; alpine snow algae communities have different assemblages
# green, orange, or red snow caused by distinct
# snow algae communities causing colored snow in alpine and subalpine summer snow 
# distinct snow algae communities across an elevational gradient
author:
  - name: Casey B. Engstrom
    institute: SFU
  - name: Kurt M. Yakimovich
    institute: SFU
  - name: Lynne M. Quarmby
    institute: SFU
    email: quarmby@sfu.ca
institute:
  - SFU: Dept. Molecular Biology and Biochemistry, Simon Fraser University, 8888 University Drive, Burnaby, BC, V5A 1S6, Canada
output: 
  bookdown::word_document2:
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=templates/scholarly-metadata.lua'
      - '--lua-filter=templates/author-info-blocks.lua'
    reference_docx: templates/word-template.docx
  # pdf_document: 
  #   fig_caption: yes
  #   number_sections: yes
csl: templates/nature-communications.csl
bibliography: refs.bib
abstract: |
  Snow algae grow in melting summer snowfields throughout polar and alpine regions, forming extensive blooms that reduce albedo and increase snow melt. Green, orange, and red patches of snow algae can harbor morphologically diverse communities, but despite their global ubiquity the community composition of snow algae blooms are poorly understood. We Illumina sequenced 18S and *rbcL* amplicons from 33 snow algae samples collected from a range of alpine and subalpine habitats in the Coast Range of British Columbia, Canada. Communities were dominated by algae closely related to *Sanguina*, *Chloromonas*, and *Chlainomonas*, as well as potentially novel taxa. Above treeline communities were dominated by *Sanguina*, and below tree line by *Chloromonas*. *Chlainomonas* was abundant in samples across all elevations. We also observed blooms of *Chloromonas krienitzii* transition from green to orange throughout the season. We found snow algae community composition was highly variable from patch to patch, even on the same mountain. Although relatively few genera dominated snow algae communities, we found high diversity at the species level. 
---

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### *Keywords*: snow algae community microbiome amplicon rbcL 18S alpine

### *Run title*: Overlapping snow algae communities






##### pagebreak

# Introduction {-}
<!-- audience, problem -->
Reduction in cryospheric albedo has become the focus of attention in recent years as an important feedback loop of global warming. Microbial growth on snow surface can reduce albedo, notably when snow algae form extensive blooms on melting summer snow[@ganey_role_2017;@lutz_biogeography_2016;@onuma_temporal_2019]. Red snow algae blooms are found on every continent, and on snow overlying sea ice[@segawa_bipolar_2018;@duval_snow_1999;@marchant_snow_1982;@yoshimura_community_1997;@vimercati_nieves_2019;@gradinger_snow_1996]. Despite their global ubiquity, the  we are only beginning to identify the various types of algae communities that cause snow algae blooms. 
<!-- why do we care if there are different communities as a climate scientist? maybe this should be a paper for snow phycologists? -->
<!-- different communities could have different ecology and life history -->
<!-- we need to define the community types to understand ecology of different blooms.  -->
<!-- it would be nice to know what the species are that form snow algae blooms, and whether different types of snow algae blooms exist.  -->
<!-- understanding of the species that cause snow algae blooms is limited to  -->

<!-- snow algae taxonomy -->
Most snow algae are green algae of class Chlorophyceae, although recent studies suggest that other classes of algae can also cause snow algae blooms[@lutz_integrated_2015;@soto_microbial_2020;@remias_two_2019]. Within the Chlorophyceae *Sanguina nivaloides* is prevalent in red snow worldwide, but many red snow sequences fall into clades which have yet to be named[@prochazkova_sanguina_2019]. Other Chlorophyceae include *Chloromonas* and the closely-related but morphologically distinct *Chlainomonas*. Currently just two species are described for *Sanguina*, while *Chloromonas* contains fourteen, and *Chlainomonas* also contains two[@matsuzaki_taxonomic_2019;@novis_two_2008]. Of the three, only *Chloromonas* has been successfully cultivated in the lab. 

<!-- microscopy suggests distinct communities -->
Microscopy suggests that snow algae blooms are in many cases are dominated by single species. In the mountains of southwestern USA and Czech Republic red snow was prevalent above treeline, dominated by cells similar in appearance to *S. nivaloides*, while below treeline green or orange snow were prevalent dominated by cells similar to *Chloromonas*[@hoham_distribution_1979;@nedbalova_ecology_2008]. Red snow dominated by *Chlainomonas* type cells were reported from slushy snow overlying alpine lakes[@novis_two_2008;@remias_ecophysiology_2016]. However, microscopic identification of snow algae is complicated by the fact that the same species can look completely different at different life stages. Isolates of *Chloromonas krienitzii* are green, bean-shaped biflagellates, but genetically identical isolated single-cells from field samples appear as orange spheres with thick cell walls, short spines, and no flagella[@matsuzaki_revisiting_2015]. Conversely, different species can look nearly identical: recent phylogenetic work shows that four similar looking cells previously referred to as *Chloromonas cf. nivalis* actually contain multiple genetically distinct clades[@matsuzaki_taxonomic_2019].

<!-- but genetic comparison, most look the same? -->
Although snow algae communities contain microscopically distinct snow algae community types, studies using 18S amplicon high-throughput sequencing (metabarcoding) suggests that snow algae communities are highly similar. 33 red snow communities across the Arctic were remarkably similar: all were dominated by uncultured Chlamydomonadaceae, with low relative abundance of *Chloromonas polyptera* and *Raphidonema nivale*[@lutz_biogeography_2016]. Another study using 18S metabarcoding of snow algae in the mountains of Japan found neighboring red and green snow patches with  highly overlapping communities, although the red snow snow contained one OTU not found in the green communities[@terashima_microbial_2017]. However, side-by-side red and green snow in Greenland contained completely different communities, and the authors concluded that red and green snow were caused by a different species[@lutz_integrated_2015].

Given how little is known about snow algae distribution and diversity, particularly in alpine habitats, our goal is to characterize the snow algae community structure in alpine and subalpine habitats. We surveyed snow algae communities from throughout the Coast Range of British Columbia, Canada using light microscopy and metabarcoding. 18S is highly conserved, making it an ideal universal barcode for eukaryotes, but at lower taxonomic levels 18S is too similar to resolve closely related taxa. To complement our 18S sequencing, we also targeted the more differentiated *rbcL* gene, coding for rubisco, thus only targeting the photosynthetic community. By using three cross-referenced metrics of relative abundance, we were able to describe the community structure with high taxonomic resolution. We found communities above treeline contained a distinct set of taxa, and instead of sharply defined community types we found something more akin to a gradient, with varying relative abundances of different snow algae taxa.   


<!--  This is an important first step before we measure habitat variables, will allow us to choose which variables to measure... -->
<!-- significance. different community, different ecology and or life history. and the snow algae are the primary producers of the community. define the study system.  -->
<!-- hypothesis: there are distinct communities of snow algae-->
<!-- methods: cell count, 18s, and rbcL amplicon sequencing -->
<!-- results: there is much overlap between communities, but many with dominant species. Some chloromonas, raphi only found with sanguina. Chlainomonas widespread. overall, no, there are not sharply delineated communities. -->
<!-- new questions raised: what habitat differences are found with different communities? is bass becking applicable here? what ecological differences, life history differences characterize the different communities?  -->



# Results {-}

We collected 309 snow algae samples from 13 mountains throughout the summer of 2018. We first detected snow algae on May 18, and we continued to sample our low elevation sites at Seymour (S) and Hollyburn (H) until the snow had completely melted out by early July. We did not detect snow algae above treeline until June 20, and snow persisted at higher elevations throughout the entire summer. 

We found morphologically distinct snow algae communities in different habitats. In shaded, forested sites we often found green or orange patches of snow, containing green or orange ovate cells with short spines resembling *Chloromonas cf. brevispina* (Fig. 1c). In forest clearings we observed low-angle linear drainage channels or runnels containing high concentrations of snow algae. In May and early June these runnels contained green snow hidden 2 to 5 cm below the white snow surface, and on subsequent visits these same sites contained orange snow at the surface, with morphologies resembling *Chloromonas krienitzii* (Fig 1b). Red snow was prevalent above treeline, dominated by red cells similar in appearance to *Sanguina nivaloides*, although we also occasionally observed green snow patches above treeline. We found red snow dominated by *Chlainomonas rubra* morphologies in unshaded sites at all elevations.

We chose 33 snow algae samples for 18S and *rbcL* amplicon Illumina sequencing. Both libraries were dominated by amplicon sequence variants (ASVs) that were assigned to Chlorophyta. Our 18S library contained 67 ASVs that were assigned to Chlorophyta and 7 ASVs to Ochrophyta, while our *rbcL* library contained 642 ASVs, of which 603 were assigned to Chlorophyta. The top genera were *Chloromonas*, *Chlainomonas*, and *Sanguina*, but most ASVs were not assigned to genus level (Fig. 2). We used t-SNE to generate operational taxonomic units (OTUs), and found many OTUs that did not correspond to any known taxa on GenBank: several OTUs related to *Chloromonas*, and one Chlamydomonadaceae OTU that was only assigned to the species level.

Above and below treeline communities contained some unique taxa, but most taxa did not exhibit a clear elevational trend (Fig. 3). *Sanguina* was the dominant taxa in most samples taken from above treeline. Additionally, high elevation sites contained several taxa not found at low elevations: an unannotated OTU of *Chloromonas*, and ASVs with a best BLAST match to *Raphidonema longiseta* (KM462868.1) were found at the three highest elevation sites. Green snow from alpine sites contained *Chloromonas* OTUs that were also found in sites below treeline. Low elevation sites contained *Chloromonas krienitzii*, but many sites were dominated by taxa that were found at all elevations, such as *Chlainomonas*. 

Cell counts, 18S, and *rbcL* datasets were generally consistent, but we did notice some discrepancies. *Chlainomonas* was more prevalent in our *rbcL* library than in cell counts, while *Sanguina* was underrepresented in *rbcL* compared to 18S and cell counts. We did not detect *Chlainomonas* in 18S. 

At runnel sites on Hollyburn and Seymour we observed snow algae blooms that changed from green to orange over the course of the summer (Fig. 1). We sequenced both green subsurface and orange surface snow at three different sites, and found both were dominated by *Chloromonas krienitzii*, although surface samples also contained some *Chlainomonas* (Supplementary _).


# Discussion

We found that the alpine snow algae community can vary widely from site to site. Different taxa were dominant at different elevations, with alpine sites dominated several distinct taxa, most notably *Sanguina*. *Chloromonas krienitzii* was dominant in mid-elevation runnels appearing as green motile cells below the snow surface, and later in the season developing orange pigment at the surface. *Chloromonas* and *Chlainomonas* were abundant across all elevations, including many taxa that are not represented in GenBank.

Our findings highlight the remaining unexplored diversity in the snow algae microbiome. Most of the diversity was only detected in *rbcL*: samples enriched in Chlamydomonadaceae E did not correspond to anything we detected in cell counts and 18S. One possibility is that this OTU represents a novel taxa of algae whose 18S is highly similar to Chloromonas. Another possibility is primer bias: perhaps this OTU was over-represented in *rbcL* and underrepresented in 18S.

Previous studies suggested that *Chlainomonas* is restricted thought to waterlogged snow overlying mountain lakes, but our results suggest that it is widespread and abundant in all types of snow algae blooms[@novis_two_2008;@prochazkova_ecophysiological_2018]. We did not detect any association with a particular habitat: only one site dominated by *Chlainomonas* was near water (S9), but other *Chlainomonas* sites were not notably wetter than the surrounding snow.

Although 18S, *rbcL*, and cell counts were generally consistent, we observed some discrepancies, notably *Sanguina* *rbcL* was underrepresented relative to 18S (Fig. 3). Primer bias could partly explain this discrepancy, because we designed our *rbcL* primers off of *Chloromonas* sequences, as *Sanguina* reference data was not available at the time. Additionally, we observed that *Sanguina* cell morphologies were more resistant to lysis than other cell types in microscopy following lysis. *Chlainomonas* was overrepresented in *rbcL* relative to cell counts, possibly due to primer bias again, or higher rRNA gene copy number typical of larger cell volumes[@fu_single_2017]. None of our 18S ASVs returned BLAST results for *Chlainomonas*, despite there being two representative GenBank sequences in our reference database (MF803743.1, MF803745.1). Possibly the *Chlainomonas* 18S sequence in our target region (V7-V8) is not different enough from closely-related *Chloromonas* to be detectable in this survey. 

Species composition is likely governed, at least in part, by different site habitat characteristics. The absence of *Sanguina* from low elevation sites could be indicative of habitat specificity: perhaps it is outcompeted at low elevation sites by taxa that are better adapted to the low light of the shade. Given the prevalence of airborne microalgae, it seems less likely that *Sanguina* is restricted in it's dispersal capabilities[@tesson_airborne_2016]. Perhaps *Chloromonas krienitzii* dominates in low-angle runnels where moisture is sufficient; alternatively, we observed that many runnels overlay streams, which could relate to the dispersal mechanism of this species. We found *Raphidonema* only in high elevation sites that were sampled late in the growing season. @stibal_growth_2005 suggests that this is a slow-growing opportunistic soil algae that blows into snow fields from surrounding dirt to colonize the snow. Perhaps sites late in the season were closer to bare ground containing source populations of this algae; alternately, perhaps this algae is slower to grow in cold temperatures on the snow, and is outcompeted by the better adapted *Sanguina*. 

Characterization of the snow algae communities allows us to define the study system, and serves as a foundation for understanding the ecology and life history of this globally significant microbiome. [...]



# Methods {-}

## Field sampling and microscopy

We collected snow algae from mountains near Vancouver, British Columbia, Canada throughout the summer of 2018. To capture the extent of snow algae diversity in our local mountains we collected as many samples as possible from different elevations, dates, mountains, and micro-habitats within the snow (S2). In total we collected 310 colored snow samples from 13 different mountains on 33 different dates from elevations between 880 m and 2150 m above sea level. In early season we sampled at lower elevations, moving uphill as the snow algae bloom progressed upwards in elevation. 

We scooped samples from visibly colored snow into 50 mL centrifuge tubes using sterile technique. To prevent melting en route to the lab we stored tubes in a bag of snow. Back in the lab, we melted each sample at room temperature on the bench and removed a 1 mL aliquot for light microscopy. Immediately after, samples were stored at -20 &deg;C for up to eight months until DNA extraction. 

We used light microscopy to characterize the cell morphologies in each sample. We prepared a slide of cells fixed in 2% gluteraldehyde, and at 400x magnification starting from the center of the slide counted up to 100 cells. We classified each cell as either *Sanguina nivaloides* [@Prochazkova2019a], *Chloromonas cf. nivalis* [@Matsuzaki2019;@Prochazkova2017], *Chloromonas cf. brevispina* [@Nedbalova2008a;@Matsuzaki2015], *Chloromonas krienitzii* [@Matsuzaki2015] *Chlainomonas rubra* [@Novis2008], or "Other" based on similarity to published photographs. 


## DNA extraction and amplicon library preparation

We selected 33 out of 310 samples for rbcL and 18s high-throughput amplicon sequencing. We chose these samples to represent the variation in date, elevation, geographic location, snow color, micro-habitat, and cell morphology. We freeze-dried these samples for up to 48 hours until until samples appeared completely desiccated. To allow water vapor to escape sample tubes while freeze drying we poked holes in each lid; as a safeguard against cross-contamination from airborne snow algae powder we wrapped tubes individually with paper towels. We mini-pestled 5 mg of sample at room temperature to physically rupture cell walls. Based on our observation that samples dominated by *S. nivaloides* cell morphologies had lower DNA yields, and had many intact cells following various lysis methods, we used 20 mg of these samples to increase our DNA yield. 

To extract DNA from the crushed cells we added 800 mL 1x CTAB extraction buffer [@ColdSp2009], 1% B-mercaptoethanol, 5 $\mu$L each of proteinase K and RNAase A, and incubated these at 65 &deg;C for 30 minutes. We centrifuged samples at 10,000 g for 3 minutes to pellet cell debris, then added 700 $\mu$L of supernatant to an equal volume of 24:1 chloroform:isoamyl alcohol. We inverted samples to mix and centrifuged again at 12,000 g for 10 minutes. We precipitated DNA by transferring the top layer to 700 $\mu$L ice-cold ethanol, which we gently mixed, and spun over Qiagen DNA columns for 30 s at 15,000 g. We washed columns twice with 70% ethanol, and finally dissolved our DNA by spinning with 50 $\mu$L of sterile TE buffer. As a negative control we processed a sterile distilled water sample alongside each batch, treating it exactly the same as the other tubes; all negative controls did not contain Qubit-detectable levels of DNA.
	
We used rbcL as a marker gene, which offers high resolution between microalgae species [@Zou2016], and has reference data for snow algae available on GenBank. We also sequenced each sample with 18S primers, due to it's coverage across a wide range of taxa, and its ubiquity in reference databases. We designed rbcL primers to target an approximately 400 bp section of this gene, based on 20 snow algae rbcL GenBank sequences from *Chloromonas* and *Chlainomonas* (GenBank accession numbers AB434272.1, LC012752.1, LC012747.1, AF517072.1, LC012738.1, LC012739.1, AB434267.1, EU030690.1, LC360494.1, AJ001878.1, AB022225.1, DQ885964.2, DQ885962.1, AJ001879.1, AB022226.1, AB022530.1, LC012751.1, AB504764.1, EU030689.1, AB101508.1). *Sanguina* sequences were not included because they were not available at the time. We designed primers using the primer design tool from the Eurofins Genomics website (www.eurofinsgenomics.eu/en/ecom/tools/pcr-primer-design/). The resultant primers were rbcL369F (5’-GAA CGT GAC AAA TTA AAC AAA-3’) and rbcL870R (5’-ACC WGA YAD ACG WAG AGC TT-3’). To target 18S we used Euk1181 (5'-TTA ATT TGA CTC AAC RCG GG-3') and Euk1624 (5'-CGG GCG GTG TGT ACA AAG G-3'), targeting the V7-V8 hypervariable regions[@Wang2014;@]. 

We constructed our amplicon library using a two-step PCR [@Meyer2010]. In the first PCR we amplified template DNA using our primers attached to a universal adapter, and in the second PCR we re-amplified that product to attach a 6 bp index to the universal adapter at the 3’ end. The first PCR total volume was 25 $\mu$L, consisting of 1 $\mu$L template, 12.5 $\mu$L Q5 high-fidelity 2X MM (New England BioLabs), 1.25 $\mu$L each of forward and reverse primer, and 9 $\mu$L of ddH2O. The second PCR was the same except we reduced our reaction volume to 20 $\mu$L by using only 5 $\mu$L of ddH2O. The cycling conditions were the same for both primer pairs for the first PCR, with an initial denaturation at 98 &deg;C for 30 s, followed by 30 cycles of 98 &deg;C for 5 s, 58 &deg;C for 10 s, and 72 &deg;C for 25 s, with a final extension at 72 &deg;C for 2 minutes. For the second indexing PCR we started with an initial denaturation at 98 &deg;C for 30 s, then 10 cycles of 98 &deg;C for 10 s, 65 &deg;C for 30 s, and 72 &deg;C for 30 s with a final denaturation of 72 &deg;C for 5 min. After each PCR, we purified using Agencourt AMPure XP kit (Beckman Coulter). We quantified final DNA concentration with Qubit (Thermo Fisher), and standardized sample concentration for pooling. The pooled library was then loaded and run on an Illumina MiSeq V3 kit.


## Bioinformatic processing

Reads were demultiplexing using CUTADAPT [@Martin2011]. Samples were filtered and trimmed, errors removed, dereplicated, pair end reads merged, and chimeras removed following the default pipeline of DADA2 [@Callahan2017].  <!-- singletons, failed samples removed, remove sequences based on negative controls, adjust for 0's, sequencing depth, -->

We assigned taxonomy for both *rbcL* and 18S ASVs using IDTaxa with threshold set to 50 [@Murali2018]. We made custom reference databases for each to include snow algae GenBank sequences. We made our rbcL database by downloading 30,865 algae rbcL sequences from GenBank with the query “rbcl[gene] AND (algae OR chlorophyta OR trebouxiophyceae) NOT (18s OR ribosomal OR psaB OR atpB) AND 300:2000[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion”. We removed ambiguous snow algae annotations based on the most recent snow algae phylogeny [@Matsuzaki2019]. Because entries labelled as *Chloromonas cf. brevispina* and *Chloromonas cf. nivalis* are polyphyletic we re-labeled these as “unassigned *Chloromonas*”. To assign 18S reads we initially ran our ASVs against SILVA [@Quast2013], and then re-ran ASVs assigned to Chlorophyta on our custom 18S snow algae database based on the following GenBank query: "(18S OR ribosomal)[gene] AND (chloromonas OR chlainomonas OR sanguina OR raphidonema OR KMY-2018) NOT (rbcL OR psaB OR atpB) AND 300:3500[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion". This database was edited to reflect the most recent snow algae taxonomy and remove ambiguous annotations [@Matsuzaki2019;@Prochazkova2018c].

As most OTU clustering algorithms are optimized for 18S data, we chose to assign high level OTUs based on the results of t-SNE clustering [@Graham-Rowe1999] (Fig. 2a). We tested this with perplexity values ranging from 1 to 100, and found comparable results with perplexity ranging from 10 to 50. We compared sample similarity using UniFrac with default settings [@Lozupone2005]. 

For software information, see Supplementary Table 9. All raw fastq files are freely available on the European Nucleotide Archive under the project accession PRJEB34539. All scripts are available at https://github.com/cengstro/bc_snow_algae_amplicon.




##### pagebreak

# Acknowledgments {-}

We wish to thank Leah Tooman (Simon Fraser University) for assistance with sequencing, and Chris Rushton (Simon Fraser University) for assistance with bioinformatics. This project was funded with a Sector Innovation Grant from Genome BC (SIP016), and a NSERC Individual Discovery Grant, both awarded to LQ. 

# Author Contributions {-}

CE, LQ, and KY designed this study. Samples collected by CE with assistance from KY and LQ. KY and CE prepped samples for sequencing. All analyses were completed by CE and KY. All authors discussed the results and contributed to the final manuscript. CE wrote the manuscript with major input from LQ and KY.

# Additional information {-}

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.





##### pagebreak

# References {-}
<div id="refs"></div> 