---
title: "Overlapping snow algae communities in the alpine and subalpine zones of British Columbia"
# who grows in the snow? multi-hued snow algae communities
# different coloured alpine and subalpine snows harbour constrasting snow algae communities
# contrasting snow algae communities in di
# not just dirty snow; alpine snow algae communities have different assemblages
# green, orange, or red snow caused by distinct
# snow algae communities causing colored snow in alpine and subalpine summer snow 
# distinct snow algae communities across an elevational gradient
author:
  - name: Casey B. Engstrom
    institute: SFU
  - name: Kurt M. Yakimovich
    institute: SFU
  - name: Lynne M. Quarmby
    institute: SFU
    email: quarmby@sfu.ca
institute:
  - SFU: Dept. Molecular Biology and Biochemistry, Simon Fraser University, 8888 University Drive, Burnaby, BC, V5A 1S6, Canada
output: 
  bookdown::word_document2:
    fig_caption: yes
    pandoc_args:
      - '--lua-filter=templates/scholarly-metadata.lua'
      - '--lua-filter=templates/author-info-blocks.lua'
    reference_docx: templates/word-template.docx
  pdf_document: 
    fig_caption: yes
    number_sections: yes
csl: templates/nature-communications.csl
bibliography: casey1.bib
abstract: |
  Snow algae grow in melting summer snowfields throughout polar and alpine regions, forming extensive blooms that reduce albedo and increase snow melt. Green, orange, and red patches of snow algae can harbour morphologically diverse communities, but despite their global ubiquity the community composition of snow algae blooms are poorly understood. We Illumina sequenced 18S and *rbcL* amplicons from 33 snow algae samples collected from a range of alpine and subalpine habitats in the Coast Range of British Columbia, Canada. Communities were dominated by algae closely related to *Sanguina*, *Chloromonas*, and *Chlainomonas*, as well as potentially novel taxa. Above treeline communities were dominated by *Sanguina*, and below tree line by *Chloromonas*. *Chlainomonas* was abundant in samples across all elevations. We also observed blooms of *Chloromonas krienitzii* transition from green to orange throughout the season. We found snow algae community composition was highly variable from patch to patch, even on the same mountain. Although relatively few genera dominated snow algae communities, we found high diversity at the species level. 
---

```{r include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


### *Keywords*: snow algae community microbiome amplicon rbcL 18S alpine

### *Run title*: Contrasting snow algae communities






##### pagebreak

# Introduction {-}
<!-- audience, problem -->
Reduction in cryospheric albedo has become the focus of attention in recent years as an important feedback loop of global warming. Microbial growth on snow surface can reduce albedo, notably when snow algae form extensive red blooms on melting summer snow[@ganey_role_2017;@lutz_biogeography_2016;@onuma_temporal_2019]. Snow algae blooms are found on every continent, and on snow overlying sea ice[@segawa_bipolar_2018;@duval_snow_1999;@marchant_snow_1982;@yoshimura_community_1997;@vimercati_nieves_2019;@gradinger_snow_1996]. Despite their global ubiquity, we are only beginning to identify the various types of algae communities that cause snow algae blooms. 
<!-- why do we care if there are different communities as a climate scientist? maybe this should be a paper for snow phycologists? -->
<!-- different communities could have different ecology and life history -->
<!-- we need to define the community types to understand ecology of different blooms.  -->

<!-- snow algae taxonomy -->
Most snow algae are green algae (Chlorophyceae), although recent studies suggest that other families of algae can also cause snow algae blooms[@lutz_omics_2015;@soto_microbial_2020;@remias_two_2019]. Within the Chlorophyceae *Sanguina nivaloides* is prevalent in red snow worldwide, but red snow also contains many other clades which have yet to be named[@prochazkova_sanguina_2019]. Other Chlorophyceaen snow algae include *Chloromonas* and the closely-related *Chlainomonas*. Currently, *Sanguina* contains two species, *Chloromonas* fourteen, and *Chlainomonas* two[@matsuzaki_taxonomy_2019;@novis_two_2008]. Only *Chloromonas* has been sucessfully cultivated in the lab. 

<!-- microscopy suggests distinct communities -->
Microscopy suggests that snow algae blooms are in many cases are dominated by single species. In the mountains of southwestern USA and Czech Republic red snow was prevelant above treeline, dominated by cells similar in appearance to *S. nivaloides*, while below treeline green or orange snow were prevalent dominated by cells similar to *Chloromonas*[@hoham_distribution_1979;nedbalova_ecology_2008]. Red snow dominated by *Chlainomonas* type cells were reported from slushy snow overlying alpine lakes[@novis_two_2008;@remias_ecophysiology_2016]. However, microscopic identification of snow algae is complicated by the fact that the same species can look completely different at different life stages. Isolates of *Chloromonas krienitzii* are green, bean-shaped biflagellates, but genetically identical isolated single-cells from field samples appear as orange spheres with thick cell walls, short spines, and no flagella[@matsuzaki_revisiting_2015]. Conversely, different species can look nearly identical: recent phylogenetic work shows that four similar looking cells previously referred to as *Chloromonas cf. nivalis* actually contain multiple genetically distinct clades[@matsuzaki_taxonomic_2019].

<!-- but genetic comparison, most look the same? -->
Although snow algae communities contain microscopically distinct snow algae community types, studies using 18S amplicon high-throughput sequencing (metabarcoding) suggests that snow algae communities are highly similar. 33 red snow communities across the Arctic were remarkably similar: all were dominated by uncultured Chlamydomonadaceae, with low relative abundance of *Chloromonas polyptera* and *Raphidonema nivale*[@lutz_biogeography_2016]. Another study using 18S metabarcoding of snow algae in the mountains of Japan found neighboring red and green snow patches with  highly overlapping communities, although the red snow snow contained one OTU not found in the green communities[@terashima_microbial_2017]. However, side-by-side red and green snow in Greenland contained completely different communities, and the authors concluded that red and green snow were caused by a different species[@lutz_omics_2015].

Given how little is known about snow algae distribution and diversity, particularly in alpine habitats, our goal is to characterise the snow algae community structure in alpine and subalpine habitats. We surveyed snow algae communities from throughout the Coast Range of British Columbia, Canada using light microscopy and metabarcoding. 18S is highly conserved, making it an ideal universal barcode for eukaryotes, but at lower taxonomic levels 18S is too similar to resolve closely related taxa. To complement our 18S sequencing, we also targeted the more differentiated *rbcL* coding for rubisco. By using three cross-referenced metrics of relative abundance, we were able to describe the community structure with high taxonomic resolution. We found communities above treeline contained a distinct set of taxa, and instead of sharply defined community types we found something more akin to a gradient, with varying relative abundances of different snow algae taxa.   


<!--  This is an important first step before we measure habitat variables, will allow us to choose which variables to measure... -->
<!-- significance. different community, different ecology and or life history. and the snow algae are the primary producers of the community. define the study system.  -->
<!-- hypothesis: there are distinct communities of snow algae-->
<!-- methods: cell count, 18s, and rbcL amplicon sequencing -->
<!-- results: there is much overlap between communities, but many with dominant species. Some chloromonas, raphi only found with sanguina. Chlainomonas widespread. overall, no, there are not sharply delineated communities. -->
<!-- new questions raised: what habitat differences are found with different communities? is bass becking applicable here? what ecological differences, life history differences characterise the different communities?  -->



# Results {-}


## Field and microscopy observations of distinct snow algae bloom types

We observed pink and green patches of snow algae in forested sites beginning on May 14. Snow algae was widespread in our study area, and we collected 309 samples from 33 dates and 13 mountains over the course of the summer (S1, S2). We observed an elevational shift in snow algae throughout the season: forested sites bloomed earlier in the season and melted out sooner, while high elevation sites did not bloom until later in the season, but persisted until the first snowfall of winter 2019 (S2).

We observed some consistent patterns between cell morphologies and field site characteristics. In forest sites, orange or green snow was typically dominated by green or yellow-orange ovate cells with short spines (*Chloromonas cf. brevispina*; Fig. 1). In snow runnels in forest clearings we observed a shift over repeated site visits; on initial visits snow runnels contained green snow hidden from sight 2 - 5 cm below the surface (Fig. 1b, S4_), on subsequent visits the surface of these runnels was orange and contained cell morphologies resembling *Chloromonas krienitzii* (Fig 1e, S3_, S4_). Red or pink snow was mostly found in open areas with high sunlight, and contained cells similar in appearance to *Sanguina nivaloides* or *Chlainomonas rubra*, in many sites mixed in with green cells of variable morphology (Fig 1). *Sanguina nivaloides* morphologies were primarily at high elevation sites, although we did observe several blooms dominated by this morphospecies in shaded forest sites. 


## Diverse *rbc*L reveals snow algae taxonomy with high-resolution

Samples were dominated by green algae (Chlorophytes), primarily the genera *Chloromonas*, *Chlainomonas*, and *Sanguina* (Fig. 2). Trebouxiophyceae (Chlorophytes) such as *Raphidonema* were present in low abundance (Fig. 2), as well as low abundance of Ochrophytes (primarily *Hydrurus*) at several low elevation sites (S5). One *rbc*L OTU defied taxonomic classification, with a best BLAST score of 88%, and top hits in both Trebouxiophyceae and Chlorophyceae (Algae H, Fig. 2). 

Our taxa-specific *rbc*L primers showed considerable variation within *Chloromonas* and the closely-related *Chlainomonas* (Fig 2, S6). We found several distinct clusters among *Chloromonas* amplicon sequence variants (ASVs) not matching any known species on GenBank (I, G, Fig. 2), but the majority of *Chloromonas* ASVs did not form distinct clusters (Other *Chloromonas*, Fig. 2). 

Our three relative abundance data sets were generally consistent. 18S, *rbc*L, and cell count data were correlated for *Sanguina* (stats). *Chloromonas* and *Chlainomonas* cell counts were correlated with *rbc*L (stats, stats); however, 18S classified samples dominated by *Chlainomonas* as containing *Chloromonas* (S5), likely due to high target region similarity (S7). However, in some samples cell morphology was a poor predictor of taxonomic composition. For example, we were surprised to find sample X1.1 contained few *Sanguina* reads (Fig. 3b, S5b) despite being dominated by red cells similar of similar morphology to *Sanguina* (Fig. 1g, S4_).


## Snow algae community composition varies with date and elevation

We found snow algae community composition was strongly influenced by elevation across all three datasets: *rbc*L, 18S, and cell count. *Sanguina* was prevalent at higher elevations, and *Chloromonas* dominant at lower elevations (Fig. 1g, Fig. 3, S5). Low elevation sites were compositionally similar, while high elevation samples showed inter-site variability (UniFrac Fig. 3a, S8). Lower elevation sites were dominated by *Chloromonas krienitzii* and *Chlainomonas rubra*, while *Sanguina nivaloides* was only abundant in samples from above treeline (Fig. 3b), along with one OTU of *Chloromonas*, and *Raphidonema* (H, I, Fig. 3b). Several sites above treeline were more similar to low elevation sites, including patches of green snow above treeline dominated by *Chloromonas* (N1.5, G1.4, Fig. 3) and red snow patches dominated by *Chlainomonas* (G1.2, P1.9, Fig. 3).

*Chlainomonas* was abundant across all elevations. Some samples were nearly completely dominated by *Chlainomonas*, but it was widespread in samples containing *Sanguina* and *Chloromonas krienitzii* as well (Fig. 3b). 

## A green snow algae turning orange at the surface

Snow algae blooms dominated by *Chloromonas krienitzii* formed in low angle runnels, often in clearings at elevations near treeline. Based on our observation that the subsurface snow was green, and surface snow turned orange through the course of the season we sampled both orange and green snow to see if they were the same species (Fig. 4a). The surface samples contained primarily orange cells resembling *Chloromonas krienitzii*, while the subsurface samples contained green cells including biflagellates and polar quadriflagellates (Fig. 4b, c). Both surface and subsurface samples were dominated by *Chloromonas krienitzii* reads (Fig. 4d). Unexpectedly, we found high relative abundance of *Chlainomonas* in surface samples, and in one subsurface sample, which we did not detect in cell counts. 






# Discussion

We found snow algae communities were highly variable from site to site, with communities at lower elevation dominated by *Chloromonas*, and higher elevation sites dominated by *Sanguina*. Microscopy studies of snow algae in USA and Slovakia found similar results, with *Chloromonas* is prevalent in forested areas and *Chlamydomonas cf. nivalis* above treeline [@Hoham1979a;@Nedbalova2008a]. Our cell counts, 18S, and *rbc*L data support this finding.

The obvious question is: what environmental factors driving those differences in community composition? Further research will be needed to determine the specific environmental factors driving community differences, but our field observations does provide some testable hypotheses. In forested areas, green cells would be shaded from damaging solar radiation, while above treeline pigmented cells of *Sanguina* and *Chloromonas* are likely protected by their astaxanthin pigment [@Gorton2003;@Holzinger2006]. This fits with our observation that green cells of *Chloromonas krienitzii* were hidden below the snow surface, and pigmented cells were more prevalent on the surface (Fig. 4); possibly pigment production is triggered as snow level melts lower, and cells are exposed to higher light at the surface. However, this does not explain why *Chloromonas krienitzii* was not found at higher elevations: our *C. krienitzii* dominated samples were taken from sites that were ~100 m from the nearest tree, and were not notably shaded by topography in any way. Anecdotally, we observed *C. krienitzii* developed over running water, so their distribution could be linked to the presence of streams below the snowpack. 

Our results show that *Chlainomonas* is an important, hitherto unrecognized component of the snow algae microbiome. Prior to this study, *Chlainomonas* were thought to only bloom in highly saturated, waterlogged snow overlying mountain lakes [@Novis2008;@Prochazkova2018c], but we found *Chlainomonas* at most sites we sampled. Only one site high in *Chlainomonas* was near water (S9), the other *Chlainomonas* dominated sites were not near water, nor was the snow noticeably mushier than the surrounding snow. This does not rule out the possibility that these sites contained high moisture at some point; further sampling is needed to determine its habitat distribution, but the broad distribution of *Chlainomonas* in most samples suggests that this taxa may be an important component of the red snow microbiome.

In contrast to typical alpine ecosystems with lower diversity at higher elevations, we observed the highest diversity at the highest elevation sites (W1.1, W1.2, P1.5, S8). These sites were our only samples collected on glaciers, and were also collected latest in the summer. These samples contained the highest proportion of *Raphidonema* (Trebouxiophyceae I, Fig. 3), which is phylogenetically distant from Chlorophyceae, and thus could contribute heavily to phylogenetic diversity metrics, however Shannon and Simpson diversity was also highest for these three samples, which only took into account the evenness of OTU abundance, so the presence of *Raphidonema* alone cannot explain the even spread of OTUs at these sites. @Stibal2005 suggest that *Raphidonema* is a slow-growing opportunistic soil algae that blows into snow fields from surrounding dirt to colonize the snow. Perhaps it's slow growth in snow means it is only found later in the season; or sites late in the season were closer to bare ground that seeded populations of this algae. 

Our use of cross-referenced 18S, *rbc*L, and cell count data uniquely allowed us to compare our findings between datasets. *Sanguina* was highly correlated between all three. One notable difference was that no reads were assigned to *Chlainomonas* from our 18S library, despite this genera being highly abundant in *rbc*L. Samples containing primarily *Chlainomonas* in our *rbc*L and cell count data were marked as containing only *Chloromonas* in 18S. At the time of publication, only two 18S *Chlainomonas* were present on GenBank (MF803743.1, MF803745.1); possibly our *Chlainomonas* sequences are different enough from the reference that they were mislabelled as *Chloromonas*, especially likely given how closely related the two genera are. Samples dominated by *Sanguina* in our 18S library contained lower abundance in *rbc*L, which is likely due to primer bias: because our custom *rbc*L primers were designed to target *Chloromonas* and *Chlainomonas* (*Sanguina* not being available at the time), this likely led to a over-representation of these taxa in our *rbc*L library. 

Our findings highlight the remaining unexplored diversity in the snow algae microbiome. One abundant OTU of algae, "Algae H" is widespread and abundant, but poorly matches all *rbc*L GenBank sequences. However, out 18S sequences did not detect any such cluster. Non-specific priming on an unsequenced gene is one possibility, another is that this represents a novel clade of snow algae, or perhaps an endosymbiotic algae within the ciliates that were abundant in many of our samples. Interestingly, the sample with the highest relative abundance of this OTU (X1.1) had red cells that were visually distinct from other *Sanguina*, with thicker cell walls that were slightly less than perfectly spherical in some cells (S4); clearly, further investigation is needed.



 Hypothetically, a positive feedback loop could permit snow algae blooms to propogate, in which snow algae melt snow, producing more liquid water that permits more snow algae growth[].
 
Our side by side comparison of snow algae communities shows that they are more diverse than previously thought, and suggests that some snow algae may be adapted to specific micro-habitats such as shade and moisture within the melting snowpack. As the different snow algae species may have varying habitat preferences, different taxa could have different life cycles that permit them to survive in their niche, thus future work to understand the biological mechanisms of snow algae blooms should take community composition into account. Different communities of snow algae will likely affect albedo and local snowpack chemistry in different ways [@Lutz2015], and so different snow algae communities could impact the microbiome and greater ecosystem in unique ways. 







# Methods {-}

## Field sampling and microscopy

We collected snow algae from mountains near Vancouver, British Columbia, Canada throughout the summer of 2018. To capture the extent of snow algae diversity in our local mountains we collected as many samples as possible from different elevations, dates, mountains, and micro-habitats within the snow (S2). In total we collected 310 colored snow samples from 13 different mountains on 33 different dates from elevations between 880 m and 2150 m above sea level. In early season we sampled at lower elevations, moving uphill as the snow algae bloom progressed upwards in elevation. 

We scooped samples from visibly colored snow into 50 mL centrifuge tubes using sterile technique. To prevent melting en route to the lab we stored tubes in a bag of snow. Back in the lab, we melted each sample at room temperature on the bench and removed a 1 mL aliquot for light microscopy. Immediately after, samples were stored at -20 &deg;C for up to eight months until DNA extraction. 

We used light microscopy to characterize the cell morphologies in each sample. We prepared a slide of cells fixed in 2% gluteraldehyde, and at 400x magnification starting from the center of the slide counted up to 100 cells. We classified each cell as either *Sanguina nivaloides* [@Prochazkova2019a], *Chloromonas cf. nivalis* [@Matsuzaki2019;@Prochazkova2017], *Chloromonas cf. brevispina* [@Nedbalova2008a;@Matsuzaki2015], *Chloromonas krienitzii* [@Matsuzaki2015] *Chlainomonas rubra* [@Novis2008], or "Other" based on similarity to published photographs. 


## DNA extraction and amplicon library preparation

We selected 33 out of 310 samples for rbcL and 18s high-throughput amplicon sequencing. We chose these samples to represent the variation in date, elevation, geographic location, snow color, micro-habitat, and cell morphology. We freeze-dried these samples for up to 48 hours until until samples appeared completely desiccated. To allow water vapor to escape sample tubes while freeze drying we poked holes in each lid; as a safeguard against cross-contamination from airborne snow algae powder we wrapped tubes individually with paper towels. We mini-pestled 5 mg of sample at room temperature to physically rupture cell walls. Based on our observation that samples dominated by *S. nivaloides* cell morphologies had lower DNA yields, and had many intact cells following various lysis methods, we used 20 mg of these samples to increase our DNA yield. 

To extract DNA from the crushed cells we added 800 mL 1x CTAB extraction buffer [@ColdSp2009], 1% B-mercaptoethanol, 5 $\mu$L each of proteinase K and RNAase A, and incubated these at 65 &deg;C for 30 minutes. We centrifuged samples at 10,000 g for 3 minutes to pellet cell debris, then added 700 $\mu$L of supernatant to an equal volume of 24:1 chloroform:isoamyl alcohol. We inverted samples to mix and centrifuged again at 12,000 g for 10 minutes. We precipitated DNA by transferring the top layer to 700 $\mu$L ice-cold ethanol, which we gently mixed, and spun over Qiagen DNA columns for 30 s at 15,000 g. We washed columns twice with 70% ethanol, and finally dissolved our DNA by spinning with 50 $\mu$L of sterile TE buffer. As a negative control we processed a sterile distilled water sample alongside each batch, treating it exactly the same as the other tubes; all negative controls did not contain Qubit-detectable levels of DNA.
	
We used rbcL as a marker gene, which offers high resolution between microalgae species [@Zou2016], and has reference data for snow algae available on GenBank. We also sequenced each sample with 18S primers, due to it's coverage across a wide range of taxa, and its ubiquity in reference databases. We designed rbcL primers to target an approximately 400 bp section of this gene, based on 20 snow algae rbcL GenBank sequences from *Chloromonas* and *Chlainomonas* (GenBank accession numbers AB434272.1, LC012752.1, LC012747.1, AF517072.1, LC012738.1, LC012739.1, AB434267.1, EU030690.1, LC360494.1, AJ001878.1, AB022225.1, DQ885964.2, DQ885962.1, AJ001879.1, AB022226.1, AB022530.1, LC012751.1, AB504764.1, EU030689.1, AB101508.1). *Sanguina* sequences were not included because they were not available at the time. We designed primers using the primer design tool from the Eurofins Genomics website (www.eurofinsgenomics.eu/en/ecom/tools/pcr-primer-design/). The resultant primers were rbcL369F (5’-GAA CGT GAC AAA TTA AAC AAA-3’) and rbcL870R (5’-ACC WGA YAD ACG WAG AGC TT-3’). To target 18S we used Euk1181 (5'-TTA ATT TGA CTC AAC RCG GG-3') and Euk1624 (5'-CGG GCG GTG TGT ACA AAG G-3'), targeting the V7-V8 hypervariable regions[@Wang2014;@]. 

We constructed our amplicon library using a two-step PCR [@Meyer2010]. In the first PCR we amplified template DNA using our primers attached to a universal adapter, and in the second PCR we re-amplified that product to attach a 6 bp index to the universal adapter at the 3’ end. The first PCR total volume was 25 $\mu$L, consisting of 1 $\mu$L template, 12.5 $\mu$L Q5 high-fidelity 2X MM (New England BioLabs), 1.25 $\mu$L each of forward and reverse primer, and 9 $\mu$L of ddH2O. The second PCR was the same except we reduced our reaction volume to 20 $\mu$L by using only 5 $\mu$L of ddH2O. The cycling conditions were the same for both primer pairs for the first PCR, with an initial denaturation at 98 &deg;C for 30 s, followed by 30 cycles of 98 &deg;C for 5 s, 58 &deg;C for 10 s, and 72 &deg;C for 25 s, with a final extension at 72 &deg;C for 2 minutes. For the second indexing PCR we started with an initial denaturation at 98 &deg;C for 30 s, then 10 cycles of 98 &deg;C for 10 s, 65 &deg;C for 30 s, and 72 &deg;C for 30 s with a final denaturation of 72 &deg;C for 5 min. After each PCR, we purified using Agencourt AMPure XP kit (Beckman Coulter). We quantified final DNA concentration with Qubit (Thermo Fisher), and standardized sample concentration for pooling. The pooled library was then loaded and run on an Illumina MiSeq V3 kit.


## Bioinformatic processing

Reads were demultiplexing using CUTADAPT [@Martin2011]. Samples were filtered and trimmed, errors removed, dereplicated, pair end reads merged, and chimeras removed following the default pipeline of DADA2 [@Callahan2017].  <!-- singletons, failed samples removed, remove sequences based on negative controls, adjust for 0's, sequencing depth, -->

We assigned taxonomy for both *rbc*L and 18S ASVs using IDTaxa with threshold set to 50 [@Murali2018]. We made custom reference databases for each to include snow algae GenBank sequences. We made our rbcL database by downloading 30,865 algae rbcL sequences from GenBank with the query “rbcl[gene] AND (algae OR chlorophyta OR trebouxiophyceae) NOT (18s OR ribosomal OR psaB OR atpB) AND 300:2000[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion”. We removed ambiguous snow algae annotations based on the most recent snow algae phylogeny [@Matsuzaki2019]. Because entries labelled as *Chloromonas cf. brevispina* and *Chloromonas cf. nivalis* are polyphyletic we re-labeled these as “unassigned *Chloromonas*”. To assign 18S reads we initially ran our ASVs against SILVA [@Quast2013], and then re-ran ASVs assigned to Chlorophyta on our custom 18S snow algae database based on the following GenBank query: "(18S OR ribosomal)[gene] AND (chloromonas OR chlainomonas OR sanguina OR raphidonema OR KMY-2018) NOT (rbcL OR psaB OR atpB) AND 300:3500[slen] NOT plasmid NOT unverified NOT mRNA NOT bacteria NOT mitochondrion". This database was edited to reflect the most recent snow algae taxonomy and remove ambiguous annotations [@Matsuzaki2019;@Prochazkova2018c].

As most OTU clustering algorithms are optimized for 18S data, we chose to assign high level OTUs based on the results of t-SNE clustering [@Graham-Rowe1999] (Fig. 2a). We tested this with perplexity values ranging from 1 to 100, and found comparable results with perplexity ranging from 10 to 50. We compared sample similarity using UniFrac with default settings [@Lozupone2005]. 

For software information, see Supplementary Table 9. All raw fastq files are freely available on the European Nucleotide Archive under the project accession PRJEB34539. All scripts are available at https://github.com/cengstro/bc_snow_algae_amplicon.




##### pagebreak

# Acknowledgments {-}

We wish to thank Leah Tooman (Simon Fraser University) for assistance with sequencing, and Chris Rushton (Simon Fraser University) for assistance with bioinformatics. This project was funded with a Sector Innovation Grant from Genome BC (SIP016), and a NSERC Individual Discovery Grant, both awarded to LQ. 

# Author Contributions {-}

CE, LQ, and KY designed this study. Samples collected by CE with assistance from KY and LQ. KY and CE prepped samples for sequencing. All analyses were completed by CE and KY. All authors discussed the results and contributed to the final manuscript. CE wrote the manuscript with major input from LQ and KY.

# Additional information {-}

The authors declare that the research was conducted in the absence of any commercial or financial relationships that could be construed as a potential conflict of interest.





##### pagebreak

# References {-}
<div id="refs"></div> 