---
title: "Supplementary materials"
output: pdf_document
---

```{r,, include=F}
knitr::opts_chunk$set(echo = FALSE)
```


```{r setup, include=FALSE}
library(here)
library(tidyverse)
library(kableExtra)
library(knitr)
library(sessioninfo)
```



```{r, include=F}
map <- knitr::include_graphics(here::here("manuscript/figs/map.png"))
field <- read_csv(here::here("data/02_tidied/tidied_field.csv"))
field_photos <- knitr::include_graphics(here::here("manuscript/figs/S3_site_field_photos/field_collage.png"))
micro_photos <- knitr::include_graphics(here::here("manuscript/figs/S4_site_micro_photos/micro.png"))
cellct <- include_graphics(here("manuscript/figs/supp/cellct/cellct.png"))
s18 <- include_graphics(here("manuscript/figs/18s_algae_rel_abund.png"))
tree <- include_graphics(here("manuscript/figs/otu_tree.png"))
tree_taxonomy <- include_graphics(here("manuscript/figs/taxonomy_tree.png"))
diversity <- include_graphics(here("manuscript/figs/diversity.png"))


map_cap <- "S1. Map showing locations of snow algae sampling locations in summer 2018."
table_cap <- "S2. Table of sample site information of sequenced snow algae samples."
field_photo_cap <- "S3. Photos of snow algae field sites."
micro_cap <- "Microphotos of snow algae samples included in sequencing run."
cellct_cap <- "Heatmap showing cell count relative abundance from 117 snow algae samples. Cb = \\emph{Chlainomonas cf. brevispina}, Ck=\\emph{Chloromonas krienitzii}, Cn=\\emph{Chloromonas cf. nivalis}, Cr=\\emph{Chlainomonas rubra}, G=green cells, O=other, Sn=\\emph{Sanguina nivaloides}"
s18_cap <- "Relative abundance barplot of 18S amplicon library"
tree_cap <- "Maximum likelihood tree of rbcL amplicon library. Colors represent OTU assignments."
tax_tree_cap <- "Maximum likelihood tree of rbcL amplicon library. Colors represent taxonomic assignments."
diversity_cap <- "A. Shannon diversity of snow algae samples, plotted against sample elevation. Smooth method = linear model. B. Faith's phylogenetic diversity plotted against sample elevation."

```

## Figures
 1. Map of sample locations
 2. Photos of field sites
 3. Microphotos of snow algae samples
 4. Plots of snow algae 18S community composition. A. All 18S, Silva assignments. B. Chlorophyta assignments with custom reference database.
 5. Maximum likelihood tree of *rbc*L amplicon
 6. Plot of base differences between Genbank 18S and *rbc*L segments
 7. Phylogenetic diversity of snow algae.

## Tables
 1. Table of sample site information
 2. Software information


```{r, fig.cap=map_cap}
# zoom in
map
```


```{r, fig.cap=field_photo_cap}
# order by date
# sizes the same (crop template)
# arrange in grid
field_photos
```


```{r, fig.cap=micro_cap}
#scale bar
# # sizes the same (crop template)
# arrange in grid
micro_photos
```

```{r s18_rel_abund, fig.cap=s18_cap}
s18 # should this go in the main fig next to the rbcl assignments?
```


```{r tree, fig.cap=tree_cap}
tree
```

```{r tax_tree, fig.cap=tax_tree_cap}
tree_taxonomy
```


```{r, fig.cap=diversity_cap}
diversity
```


```{r, (ref:cellct_cap)}
cellct
```




```{r}
field %>% 
   filter(have_seq==T) %>% 
   mutate(notes = if_else(sample_id %in% c("pan18.ro","bre18.rd"), "collected from rock",NA_character_)) %>% 
   select(alias, date, elev_m, lat, lon,color_snow, habitat, notes) %>% 
   arrange(date) %>% 
   kable(caption = table_cap) %>% 
   kable_styling(bootstrap_options = "striped")
```



```{r}
my_session_info <- sessioninfo::session_info()
  
my_session_info$platform %>% 
  unlist() %>% 
  enframe() %>% 
  kable(caption = "Session information") %>% 
  kable_styling(bootstrap_options = c("striped", "condensed"), full_width = F, position = "left")
```


```{r}
software <- my_session_info$packages %>% 
  as_tibble() %>% 
  select(package, loadedversion) %>% 
  dplyr::rename(version = loadedversion)

# cedar_info <- tibble(package = c(""),
#                      version = c(""))

# software <- software %>% 
#   bind_rows(cedar_info)

software %>% 
  kable(caption=" R software package versions used") %>% 
  kable_styling(bootstrap_options = c("striped", "condensed"), full_width = F, position = "left")
```

