---
title: "ASV visualization"
---


```{r libs, echo=F}
library(tidyverse)
library(here)
library(DECIPHER)
library(ggforce)
library(Rtsne)
library(ape)
library(ggtree)
library(treeio)
library(dbscan)
library(ggthemes)
library(ggpubr)
# library(plotly)
# library(crosstalk)
# library(DT)
```

```{r set}
theme_set(theme_few())
set.seed(123)
```

```{r read}
rbcl_asv_id_key <- read_csv(here("output","keys","rbcl_asv_key.csv"))
rbcl_rel_abund <- read_csv(here("data","clean","rbcl_rel_abund_clean.csv"))
idtax <- read_csv(here("data","clean","rbcl_taxonomy_clean.csv"))
```

Align rbcL ASVs
```{r align}
# view alignments, make distance matrix
asv.dss <- rbcl_asv_id_key$asv %>%
  DNAStringSet()
names(asv.dss) <- rbcl_asv_id_key$asv_id
asv.aligns <- asv.dss %>% 
  DECIPHER::AlignSeqs(verbose=F)

asv.aligns %>% DECIPHER::BrowseSeqs(highlight = T) # 600 has a ~70 bp insert, 607 ~45 bp deletion, 508 ~110 bp deletion
length(asv.aligns)
```

Make distance matrix
```{r dists}
dists <-  asv.aligns %>% 
  DECIPHER::DistanceMatrix(type="matrix", correction = "JC")
```

t-SNE with distance matrix
```{r tsne}
perplexity <- 30

tsne <- dists %>% 
  Rtsne(pca=F, perplexity = perplexity)

tsne <- tsne$Y %>% 
  as_tibble()

ggplot(tsne, aes(x=V1,y=V2))+
  geom_point()
```

Add taxonomy assignments, see if t-SNE agrees with the reference-based clustering of IDTaxa
```{r genus_tsne}
tsne <- tsne %>% 
  add_column(asv_id = colnames(dists)) %>% # label each point by it's ASV ID
  left_join(idtax, by="asv_id")

ggplot(tsne, aes(x=V1,y=V2))+
  geom_point(aes(color = genus))
```
Seperates out the Sanguia, Chlainomonas. Lumps all Trebouxiophyceae together. Lots of clustering within Chloromonas, some big clusters with only one point assigned to Chloromonas. 

check at species level
```{r sp_tsne}
ggplot(tsne, aes(x=V1,y=V2))+
  geom_point(aes(color = species))
```


Make it look nice, show both genus and species level assignments
```{r tsne-plot1}

# make new categorical to tell ggplot what taxonomy to plot
# include both species and genus level assignments
# lump low abundance Trebouxiophyceae
treb <- c("Myrmecia","Trebouxia","Coccomyxa","Dictyochloropsis","Diplosphaera")
tsne <- tsne %>% 
    mutate(display_taxa = case_when(genus %in% treb ~"other Trebouxiophyceae",
                                  genus == "Stichococcus" ~ "Stichococcus",#########not sure why I need this...
                                  id_level == "species" ~ assignment,
                                  id_level == "genus" ~ assignment))


# order
taxa.order <- c("Chloromonas","Chloromonas_muramotoi","Chloromonas_hohamii","Chloromonas_krienitzii","Chloromonas_Group_B","Raphidonema_nivale","Stichococcus","other Trebouxiophyceae","Chlainomonas", "Chlainomonas_rubra","Sanguina","Sanguina_aurantia","Sanguina_nivaloides" )

tsne_plot <- ggplot(tsne, aes(x=V1, y=V2)) +
  # plot points not assigned to genus or species
  geom_point(data = filter(tsne, is.na(display_taxa)), 
             color = "grey",alpha = 0.8, size = 3) +
  # color points that were assigned to genus or species
  geom_point(data = filter(tsne, !is.na(display_taxa)), 
             aes(color = display_taxa %>% fct_relevel(taxa.order), alpha = 0.8), size = 3)  
tsne_plot
```

cluster t-SNE into OTUs
```{r otu-cluster}
eps <- 4

tsne.mat <- tsne %>% 
  select(1,2) %>% 
  as.matrix()

dbscan.out <- dbscan::dbscan(tsne.mat, eps=eps) # eps=4 gives best results

tsne <- tsne %>% 
  add_column(otu = as.character(dbscan.out$cluster)) # tsne is in order of asv_id

ggplot(tsne, aes(x=V1, y=V2)) +
  geom_point(aes(color = otu))
```

manually add *Chloromonas krienitzii* cluster, since this is highly abundant, and split *Chlainomonas* into two, based on the observation that the two clades of *Chlainomonas* seperate out by elevation
```{r edit-otus}
tsne <- tsne %>%
  mutate(otu = case_when((V2 < (-5) & V1 < -17) ~ "9", # chlain "high"
                         (V2 < 14  & V2 > 8 & V1 > (-20) & V1 < (-12)) ~ "10",# krienitzii
                         TRUE ~ otu))

tsne %>% 
  select(display_taxa, otu) %>% 
  distinct() %>% 
  arrange(otu)

# name each OTU
otu_names <- tibble(otu = as.character(1:10), otu_name = c("Chloromonas spp. C","Sanguina A","Chlainomonas B", "Chloromonas muramotoi E","Chloromonas G","Chloromonas I","Algae H","Trebouxiophyceae J","Chlainomonas F","Chloromonas krienitzii D"))

tsne <- tsne %>% 
  left_join(otu_names, by="otu")

asv_otu_key <- tsne %>% 
  select(asv_id, otu, otu_name)
  
tsne_plot <- tsne_plot +
  ggforce::geom_mark_ellipse(data = tsne, aes(fill=otu_name, label=otu_name))

# set



# extract the legend
tsne_plot %>% 
  get_legend() %>% 
  as_ggplot()
ggsave(here("figs","asv-cluster","asv-legend.pdf"), height = 7, width = 3)

# plot and save the main plot
tsne_plot +
  theme(legend.position = "none")
ggsave(here("figs","asv-cluster","asv-cluster.pdf"), height = 5, width = 5)
```

# Phylogenetic tree

Write fasta for IQTree web serv
```{r}
asv.aligns %>% writeXStringSet(here("output","tree","rbcl","rbcl_aligns.fasta"))
```

Run IQtree
```{r}
rbcl.tree <- read.tree(here("output","tree","rbcl","asv613.fasta.treefile"))

idtax %>% filter(species == "Raphidonema_nivale")
# re-root tree
rbcl.tree <- rbcl.tree %>% root(outgroup = "559", resolve.root=T)

rbcl_ggtree <- ggtree(rbcl.tree)

# parse bootstrap values
rbcl_ggtree_data <- rbcl_ggtree$data %>% 
  filter(isTip == FALSE) %>% 
  separate(label, c("boot", "bayes"), sep = "/") %>% #split label into boots and bayes vals
  filter(boot > 80) %>%
  mutate(boot = round(as.numeric(boot), digits = 0),
         insignificant_edge = if_else(boot > 80, "black", "red"))# round boot to nearest integer

# Plot trees
rbcl_ggtree %<+% select(tsne, asv_id, display_taxa) + # attach taxonomic assignment for tip label
  geom_tippoint(aes(color=display_taxa, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

rbcl_ggtree %<+% asv_otu_key + # attach OTU
  geom_tippoint(aes(color=otu_name, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

# color edge red if boot < 80
# ggtree_conf <- rbcl_ggtree$data %>% 
#   separate(label, c("boot", "bayes"), sep = "/") %>% #split label into boots and bayes vals
#   mutate(insignificant_edge = case_when((boot < 80 & isTip == FALSE)~"dotted", 
#                                         TRUE~"solid"),
#          asv_id = case_when(isTip == TRUE~boot,
#                             TRUE~NA_character_)) %>% # round boot to nearest integer
#   left_join(otu_tbl, by="asv_id") 



# rbcl_ggtree %<+% ggtree_conf +
#   aes(linetype = I(insignificant_edge)) +
#   geom_tippoint(aes(color=otu),alpha=0.7, size=3) + #
#   theme(legend.position = c(0.2,0.7), 
#         legend.title = element_blank(), 
#         legend.key = element_blank()) 
# ggsave("figs/ggtree.pdf", height = 10, width=7.5)
```




```{r output}

write_csv(asv_otu_key, here("output","keys","asv_otu_key.csv"))
```


# TO do
- add MDS
- point size proportional to relative abundance
- ggtree
- show different values of perplexity yield comparable results
