---
title: "Just algae 18S community comparison"
output: html_notebook
---

```{r libs, include=F}
library(tidyverse)
library(here)
library(GUniFrac)
library(vegan)
library(vegalite)
library(fossil)
library(picante)

# ggplot themes
library(ggthemes)
library(ggdark)
```


```{r, include=F}
rel_abund <- read_csv(here("data/03_filtered/18s_algae/18s_algae.filtered_cleaned_rel_abund.csv"))
algae_taxonomy <- read_csv(here("output/algae_18s_assignments_snow_algae_genbank.csv"))
algae_select_taxonomy <- read_csv(here("output/algae_18s_assignments_w_select_snow_algae.csv"))

tree_18s <- read.tree(here("output/tree/algae_18s/algae_18s_seqs.fasta.treefile"))
field <- read_csv(here("data/02_tidied/tidied_field.csv"))
```




```{r, include=FALSE}
taxonomy <- algae_taxonomy

algae_percent_rel_abund <- rel_abund %>% 
  # calculate rel abund as % of Chlorophyta reads
  group_by(sample_id) %>% 
  mutate(rel_abund = n_reads/sum(n_reads)) %>% 
  ungroup()

best_assignment_rel_abund <- algae_percent_rel_abund %>% 
  # calculate relative abundance per genus
  left_join(taxonomy, by="asv_id") %>% 
  group_by(sample_id, best_assignment) %>% 
  summarise(rel_abund = sum(rel_abund)) %>% 
  ungroup() %>% 
  left_join(field, by="sample_id")


barplot <- ggplot(best_assignment_rel_abund, aes(x=alias %>% fct_reorder(elev_m), y=rel_abund)) +
  geom_bar(aes(fill=best_assignment), stat="identity") + # classic stacked relative abundance plot
  coord_flip() +
  labs(title="Assignment with unedited genbank downloads",
       fill = "18S taxonomic assigmnent",
       y="Relative abundance out of Chlorophyta reads",
       x="Sample ID (in order of increasing elevation)")
barplot


# desired categories: lump C nivalis in Chloromonas, lump 

```

```{r}
taxonomy <- algae_select_taxonomy

best_assignment_rel_abund <- algae_percent_rel_abund %>% 
  # calculate relative abundance per genus
  left_join(taxonomy, by="asv_id") %>% 
  group_by(sample_id, best_assignment) %>% 
  summarise(rel_abund = sum(rel_abund)) %>% 
  ungroup() %>% 
  left_join(field, by="sample_id")


barplot_select <- ggplot(best_assignment_rel_abund, aes(x=alias %>% fct_reorder(elev_m), y=rel_abund)) +
  geom_bar(aes(fill=best_assignment), stat="identity") + # classic stacked relative abundance plot
  coord_flip() +
  labs(title="Assignment with select database",
       fill = "18S taxonomic assigmnent",
       y="Relative abundance out of Chlorophyta reads",
       x="Sample ID (in order of increasing elevation)")
barplot_select

```

Try with consensus sequence for each known snow algae species. 

Takehome: 
  - X1.1 has high Trebouxiophyceae of some sort, Koliellaceae
  - W1.1 Chlorogonium (Haematococcaceae), 
  - similar patterns otherwise as rbcL, except no Chlainomonas
  


```{r}
algae_percent_rel_abund %>% 
  filter(sample_id == "sax18.01") %>% 
  arrange(-rel_abund)
```

top two ASVs in sax18.01 blast as koliellaceae and chloromonas




# Unifrac Algae 18s

convert to wide format, asvs as cols
```{r unifrac}
wide_18s_ra <- clean_18s_ra %>% 
  select(asv_id, sample_id, rel_abund) %>% 
  filter(asv_id %in% tree_18s$tip.label) %>% # Unifrac needs relative abundance table to match the tips on the tree
  pivot_wider(names_from = asv_id, values_from = rel_abund, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("sample_id")
head(wide_18s_ra)
```

Choose a Trebouxiophyceae to root tree by
```{r}
root <- silva %>% 
  filter(class == "Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(1)
  
tree_18s <- tree_18s %>% root(outgroup = root, resolve.root=T)
```


run Unifrac
```{r} 
  
set.seed(123)
unifrac_out <- wide_18s_ra %>% 
  GUniFrac::GUniFrac(tree_18s, alpha=c(0, 0.5, 1)) # ignore error message, we removed some otus from our tree, they should be automatically discluded

# select the weighted UniFrac matrix
weighted_unifrac_distm <- unifrac_out$unifracs[, , "d_1"]
dim(weighted_unifrac_distm)
```

Ordination (NMDS) of 18s UniFrac, check stress value
```{r}
mono <- metaMDS(weighted_unifrac_distm)
iso <- metaMDS(weighted_unifrac_distm, engine = "isoMDS")

mono
iso


```

stress too low to plot... with the few algae asvs, perhaps it all looks too similar

try with all 18s data, not just algae


    ```{r}
    unifrac_plot_data <- nmds %>%
      scores() %>%
      as_tibble(rownames = "sample_id") %>%
      left_join(field, by="sample_id")
    
    # plot unifrac, colored by elevation
    uf_plot <- ggplot(unifrac_plot_data, aes(x=NMDS2, y=NMDS1)) +
      geom_point(aes(color=elev_m), size=4) +
      scale_color_gradientn(colours = terrain.colors(8)[1:8]) +
      labs(color="Elevation (m)") #+
      # geom_text(data = unifrac_plot_data %>%   # add sample ID labels
                  # left_join(sample_aliases, by="sample_id"),
                # aes(label = alias), size=3,
                # position = position_nudge(y = -0.025))  ######### change to alias for final version. edit overlap txt in svg
    uf_plot
```



```{r}
ggsave(here("manuscript/figs/18s_algae_rel_abund.png"), plot = barplot)
```

