---
title: "Community comparison"
output: html_notebook
---

```{r libs, echo=F}
library(tidyverse)
library(here)
library(GUniFrac)
library(vegan)
library(vegalite)
library(fossil)
library(picante)

# ggplot themes
library(ggthemes)
library(ggdark)
```

```{r set}
# theme_set(ggdark::dark_theme_classic()) # for presentation figures
theme_set(theme_classic())

```


```{r input, echo=F}
rbcl_rel_abund <- read_csv(here("data","clean","rbcl_rel_abund_clean.csv"))
idtax <- read_csv(here("data","clean","rbcl_taxonomy_clean.csv"))
field <- read_csv(here("data","clean","field_clean.csv"))

rbcl_asv_id_key <- read_csv(here("output","keys","rbcl_asv_key.csv"))
rbcl_asv_otu_key <- read_csv(here("output","keys","asv_otu_key.csv"))
sample_aliases <- read_csv(here("data","sample_aliases.csv"))
rbcl.tree <- read.tree(here("output","tree","rbcl","asv613.fasta.treefile"))
```

Unifrac rbcL
```{r unifrac}
rbcl.asv.rel.abund.wide <- rbcl_rel_abund %>% 
  select(asv_id, sample_id, rel_abund) %>% 
  pivot_wider(names_from = asv_id, values_from = rel_abund, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("sample_id")

# rooted tree required, root by one of my Raphidonema asvs
rbcl.tree <- rbcl.tree %>% root(outgroup = "559", resolve.root=T)

set.seed(123)
unifracs <- rbcl.asv.rel.abund.wide %>% 
  GUniFrac::GUniFrac(rbcl.tree, alpha=c(0, 0.5, 1))
unifrac <- unifracs$unifracs[, , "d_1"]


# plot results as NMDS
nmds <- metaMDS(unifrac, autotransform = F)
nmds # stress
unifrac_plot_data <- nmds %>% 
  scores() %>% 
  as_tibble(rownames = "sample_id") %>% 
  left_join(field, by="sample_id")

# plot unifrac, colored by elevation
uf_plot <- ggplot(unifrac_plot_data, aes(x=NMDS2, y=NMDS1)) +
  geom_point(aes(color=elev_m), size=4) +
  scale_color_gradientn(colours = terrain.colors(8)[1:8]) +
  labs(color="Elevation (m)") #+
  # geom_text(data = unifrac_plot_data %>%   # add sample ID labels
              # left_join(sample_aliases, by="sample_id"),
            # aes(label = alias), size=3,
            # position = position_nudge(y = -0.025))  ######### change to alias for final version. edit overlap txt in svg
uf_plot
```

# rbcL relative abundance plot
```{r barplot, include=F}
rbcl_rel_abund_tbl <- rbcl_rel_abund %>%
  left_join(rbcl_asv_otu_key, by="asv_id") %>%
  left_join(field, by="sample_id") %>%
  left_join(sample_aliases, by="sample_id") %>%
  select(alias, rel_abund, otu_name, elev_m) %>%
  group_by(alias, otu_name, elev_m) %>%
  summarise(rel_abund = sum(rel_abund)) %>%
  ungroup()

rbcl_rel_abund_tbl %>% distinct(otu_name)

# sanity check samples rel abund sum to 1
rbcl_rel_abund_tbl %>%
  group_by(alias) %>%
  summarise(sum(rel_abund))

# order OTUs
otu.order <- c("Sanguina A","Chloromonas I","Chloromonas G","Chloromonas muramotoi E","Chloromonas krienitzii D","Chloromonas spp. C","Chlainomonas B","Chlainomonas F","Algae H", "Trebouxiophyceae J")

# specify colors for OTUs
#           Sang         Chlo I    Chl G      Chl mura    Chl kri   chlor sp  chlain B    chlain F   Alg H     Treb J
colors <- c("#e31a1c", "#a6cee3", "#1f78b4", "#b2df8a", "#33a02c", "#00BFC4", "#cab2d6", "#6a3d9a", "#fdbf6f", "#ff7f00")

stacked_barplot <- ggplot(rbcl_rel_abund_tbl,
                          aes(x = alias %>% fct_reorder(elev_m),
                              y = rel_abund,
                              fill=otu_name %>% fct_relevel(otu.order))) +
  geom_bar(position="stack",stat="identity") +
  coord_flip() +
  xlab("Sample ID") +
  ylab("Relative abundance") +
  labs(fill = "OTU") +
  scale_fill_manual(values = colors)
stacked_barplot

# elevation sidebar
ggplot(rbcl_rel_abund_tbl,
       aes(x = alias %>% fct_reorder(elev_m),
           y = elev_m,
           fill = elev_m)) +
  geom_bar(stat="identity") +
  coord_flip() +
  scale_fill_gradientn(colors = terrain.colors(8)[1:7])
```

rbcL heatmap
```{r heatmap, fig.height=6, fig.width=6}
                                                 
heatmap <- ggplot(rbcl_rel_abund_tbl, aes(y = alias %>% fct_reorder(elev_m),
                               x = otu_name %>% fct_relevel(otu.order))) +  
  geom_tile(aes(alpha = rel_abund, fill = otu_name %>% fct_relevel(otu.order)), color = "white") + 
  scale_alpha(range = c(0.1, 1)) +
  ylab("Sample ID") +
  xlab("OTU") +
  labs(fill = "OTU", alpha="Relative abundance") +
  scale_x_discrete(position = "bottom") +
  theme(axis.text.x = element_text(hjust=1,angle=45)) +
  guides(fill=FALSE)
heatmap
```


# Diversity

Chao 

```{r chao1}
# asv-based chao1
rbcl_rel_abund_matrix <- rbcl_rel_abund %>%
  pivot_wider(id_cols = -rel_abund, names_from = sample_id, values_from = n_reads, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("asv_id") %>% 
  as.matrix() %>% 
  t()
  
# fossil::chao1(rbcl_rel_abund_matrix)





# re-root tree
rbcl.tree <- rbcl.tree %>% root(outgroup = "559", resolve.root=T)
  
faiths <- picante::pd(rbcl_rel_abund_matrix, rbcl.tree) %>% 
  # attach elevation data
  rownames_to_column("sample_id") %>% 
  left_join(field, by="sample_id") %>% 
  select(sample_id, PD, date, elev_m)
 # plot PD as function of elevation
faiths

faith_date_plot <- ggplot(faiths, aes(y=PD, x=date))+
  geom_point() +
  geom_smooth(method = "lm")+
  ylab("Faith's phylogenetic diversity (PD)") +
  xlab("Date")
faith_date_plot

lm_out <- lm(PD ~ date + elev_m, data = faiths)
summary(lm_out)

# just elev
lm_out1 <- lm(log(PD) ~ elev_m, data = faiths) %>% # log transform to normalize
  broom::tidy()

elev_slope <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(estimate)
elev_p <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(p.value)

ggplot(faiths, aes(y=log(PD), x=elev_m))+
  geom_point() +
  geom_smooth(method = "lm")+
  ylab("log Faith's phylogenetic diversity (PD)") +
  xlab("Elevation (m)")

# just date
lm_out2 <- lm(PD ~ date, data = faiths) %>% 
  broom::tidy()

date_slope <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(estimate)
date_p <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(p.value)

# check resids
hist(resid(lm_out1))

# calculate co-linearity
car::vif(lm_out)
```

Later dates and higher elevations had higher within-sample diversity. Phylogenetic diversity (PD) [@Faith1992] increased linearly with date by `r date_slope` per day (p = `r date_p`), and log-transformed PD increased by `r exp(elev_slope)` per metre gain in elevation (p = `r elev_p`). (Supplemental Figure X)


```{r output}
ggsave(plot = uf_plot, filename = here("figs","rbcl_unifrac_nmds.png"), height = 4, width = 5)

path <- here("figs/heatmap_dark.png")
ggsave(plot=heatmap, filename=path, height = 8, width = 6)

ggsave(plot=faith_date_plot, filename = here("figs/faith_diversity_date.png"), height = 7, width = 7)
```