---
title: "rbcL ASV clustering"
output: html_notebook
---


```{r libs, include=F}
library(tidyverse)
library(here)
library(DECIPHER)
library(ggforce)
library(Rtsne)
library(ape)
library(ggtree)
library(treeio)
library(dbscan)
library(ggrepel)

library(ggpubr)
library(ggthemes)
library(ggdark)
```

```{r read, include=F}

rel_abund <- read_csv(here("data/03_filtered/rbcl/filtered_cleaned_rbcl_rel_abund.csv"))
taxonomy <- read_csv(here("data/02_tidied/tidied.rbcl_taxonomy.csv"))

aligns <- readDNAStringSet(here("output/alignments/rbcl_aligns_in_main.fasta"))
tree <- read.tree(file=here("output/tree/rbcl/asv613.fasta.treefile")) # this is the tree with only the ASVs that appear in the filtered rel abund dataset (not including protocol stuff, etc)
```

```{r asv_distance_matrix, include=F}
BrowseSeqs(aligns, highlight = T)

dists <-  aligns %>% 
  DECIPHER::DistanceMatrix(type="matrix", correction = "JC")
```

```{r tsne_1, echo=F}
perplexity <- 30

set.seed(123)
tsne_full_output <- dists %>% 
  Rtsne(pca=F, perplexity = perplexity)

tsne_pts <- tsne_full_output$Y %>% 
  as_tibble()

ggplot(tsne_pts, aes(x=V1,y=V2))+
  geom_point() +
  labs(title="rbcL ASV clustering with t-SNE")
```

```{r genus_tsne, echo=F}
tsne_pts_taxonomy <- tsne_pts %>% 
  add_column(asv_id = colnames(dists)) %>% # label each point by it's ASV ID
  left_join(taxonomy, by="asv_id")

ggplot(tsne_pts_taxonomy, aes(x=V1,y=V2))+
  geom_point(aes(color = genus)) +
  labs(title = "Points labeled by genus")
```



```{r sp_tsne, echo=F}
ggplot(tsne_pts_taxonomy, aes(x=V1,y=V2))+
  geom_point(aes(color = species)) +
  labs(title="Pts labeled by species")
```


Make it look nice, show both genus and species level assignments
```{r tsne-plot1, echo=F}

taxa.order <- c("Chloromonas","Chloromonas_muramotoi","Chloromonas_hohamii","Chloromonas_krienitzii","Chloromonas_Group_B","Raphidonema_nivale","Stichococcus","other Trebouxiophyceae","Chlainomonas", "Chlainomonas_rubra","Sanguina","Sanguina_aurantia","Sanguina_nivaloides" )

ggplot(tsne_pts_taxonomy, aes(x=V1, y=V2)) +
  # plot points not assigned to genus or species
  geom_point(data = filter(tsne_pts_taxonomy, is.na(plot_taxa)), 
             color = "grey", alpha = 0.8) +
  # color points that were assigned to genus or species
  geom_point(data = filter(tsne_pts_taxonomy, !is.na(plot_taxa)), 
             aes(color = plot_taxa %>% fct_relevel(taxa.order)), 
             alpha = 0.8) +
  labs(title = "Points colored by both genus and species", color="ASV taxonomic assignment")
```


```{r otu-cluster, echo=F}
eps <- 4

tsne_mat <- tsne_pts_taxonomy %>% 
  select(1,2) %>% 
  as.matrix()

set.seed(123)

dbscan_out <- dbscan::dbscan(tsne_mat, eps=eps) # eps=4 gives best results

tsne_taxa_otu <- tsne_pts_taxonomy %>% 
  add_column(otu = as.character(dbscan_out$cluster)) # tsne is in order of asv_id

ggplot(tsne_taxa_otu, aes(x=V1, y=V2)) +
  geom_point(aes(color = otu)) +
  labs(title="dbscan clustering")
```

Manually edit the clustering to show *Chloromonas krienitzii* on relative abundance barplot

```{r edit-otus, echo=F}
tsne_edited_otus <- tsne_taxa_otu %>%
  mutate(otu = case_when((plot_taxa == "Chloromonas_krienitzii") ~ "9",
                         TRUE ~ otu))

otus <- tsne_edited_otus %>% 
  select(asv_id, otu)

# get names for each otu
rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  left_join(taxonomy, by="asv_id") %>% 
  distinct(otu, plot_taxa) %>% 
  arrange(otu)

# get rel abund of each otu
rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  group_by(otu) %>% 
  summarise(rel_abund = sum(rel_abund)) %>% 
  arrange(-rel_abund) %>% 
  select(-rel_abund) %>% 
  add_column(letter = LETTERS[1:9])

# name each OTU, lettered in order of rel abund
otu_names <- tibble(otu = as.character(1:9), 
                    otu_name = c("Chloromonas spp. C",
                                 "Sanguina B",
                                 "Chlainomonas A",
                                 "Chloromonas muramotoi D",
                                 "Chloromonas E",
                                 "Chloromonas G",
                                 "Algae F",
                                 "Trebouxiophyceae H",
                                 "Chloromonas krienitzii"))

otus_asv_name <- otus %>% 
  left_join(otu_names, by="otu") %>% 
  select(asv_id, otu_name)
```


```{r echo=F, fig.width=8, fig.height=6}
tsne_edited_otus %>% 
  select(plot_taxa, otu) %>% 
  distinct() %>% 
  arrange(otu)





# make a key for downstream use (write to file at end of script)
asv_otu_key <- tsne_otu_names %>% 
  select(asv_id, otu, otu_name)
  
ggplot(tsne_otu_names, aes(x=V1, y=V2)) +
  ggforce::geom_mark_ellipse(aes(fill=otu_name, label=otu_name), alpha=0.3, show.legend = FALSE) +
  # plot points not assigned to genus or species
  geom_point(data = filter(tsne_otu_names, is.na(plot_taxa)), 
             color = "grey",alpha = 0.8, size = 3) +
  # color points that were assigned to genus or species
  geom_point(data = filter(tsne_otu_names, !is.na(plot_taxa)), 
             aes(color = plot_taxa %>% fct_relevel(taxa.order)), alpha = 0.8, size = 3) +
  labs(title = "Points colored by both genus and species", color="ASV taxonomic assignment") +
  theme_few() 
```

```{r include_rel_abund, echo=F}
total_rel_abund <- rel_abund %>% 
  group_by(asv_id) %>% 
  summarise(global_rel_abund = sum(rel_abund))

tsne_rel_abund <- tsne_otu_names %>% 
  left_join(total_rel_abund, by="asv_id")

final_tsne <- ggplot(tsne_rel_abund, aes(x=V1, y=V2)) +
  ggforce::geom_mark_ellipse(aes(fill=otu_name, label=otu_name), 
                             show.legend = FALSE, 
                             alpha=0.2, 
                             linetype="blank", 
                             expand = unit(2, "mm")) +
  # plot points not assigned to genus or species
  geom_point(data = filter(tsne_rel_abund, is.na(plot_taxa)), 
             aes(size = global_rel_abund),
             color = "grey",alpha = 0.5) +
  # color points that were assigned to genus or species
  geom_point(data = filter(tsne_rel_abund, !is.na(plot_taxa)), 
             aes(color = plot_taxa %>% fct_relevel(taxa.order),
                 size = global_rel_abund), 
             alpha = 0.5) +
  labs(color="ASV taxonomic assignment", tag="A") +
  ggthemes::theme_few() +
  scale_size(range = c(1, 6), guide="none") + ################################################# not changing the scale size
  theme(legend.position = "none")
final_tsne
```


```{r asv_mds, echo=F}
mds_coords <- dists %>% 
  cmdscale() %>% # ok to use euclidian dist, no null values when comparing relatedness
  as_tibble() %>% 
  add_column(asv_id = colnames(dists), .before=1)

mds_asv_plot <- mds_coords %>% 
  left_join(taxonomy, by="asv_id") %>% 
  left_join(asv_otu_key, by="asv_id") %>% 
  left_join(total_rel_abund, by="asv_id") %>% 
  filter(!(asv_id %in% c("639", "534", "646"))) # remove Trebouxiophyceae outliers

mds <- ggplot(mds_asv_plot, aes(V1, V2)) + 
  ggforce::geom_mark_ellipse(data = mds_asv_plot %>% 
                               filter(otu_name != "Trebouxiophyceae J"), # dont plot the ellipse around other treb
                             aes(fill = otu_name, label=otu_name),
                             show.legend = FALSE, 
                             alpha=0.2, 
                             linetype="blank",
                             expand = unit(2, "mm")) +
  geom_point(aes(alpha = 0.7, color = plot_taxa, size = global_rel_abund)) +
  scale_size(range = c(1, 6), guide="none") +
  scale_alpha(guide="none")+
  theme_few() +
  labs(color="ASV taxonomic assignment", tag="B")
  # geom_text(aes(label = asv_id)) # too see outliers, asv_id=c("639", "534")
mds
```



# Phylogenetic tree


Run IQtree
```{r tree, fig.width=8, fig.height=20}
# root tree with Trebouxiophyceae
root <- taxonomy %>% 
  filter(class=="Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(asv_id)

rbcl.tree <- rbcl.tree %>% root(outgroup = root, resolve.root=T)



rbcl_ggtree <- ggtree(rbcl.tree)

# parse bootstrap values
rbcl_ggtree_data <- rbcl_ggtree$data %>% 
  filter(isTip == FALSE) %>% 
  separate(label, c("boot", "bayes"), sep = "/") %>% #split label into boots and bayes vals
  filter(boot > 80) %>%
  mutate(boot = round(as.numeric(boot), digits = 0),
         insignificant_edge = if_else(boot > 80, "black", "red"))# round boot to nearest integer

# Plot trees

# tips colored by taxonomy
taxonomy_tree <- rbcl_ggtree %<+% taxonomy + 
  geom_tippoint(aes(color=plot_taxa, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

# clades annotated by OTU
otu_tree <- rbcl_ggtree %<+% asv_otu_key + 
  geom_tippoint(aes(color=otu_name, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

otu_tree

```

todo: 
- better color choices
- circular tree


```{r output}
write_csv(otus_asv_name, path = here("output/asv_otu_key.csv"))

ggsave(filename = here("manuscript/figs/2_taxonomy/tsne.png"), plot=final_tsne, height = 6, width = 8)
ggsave(filename = here("manuscript/figs/2_taxonomy/mds.png"), plot=mds,  height = 6, width = 8)
ggsave(filename = here("manuscript/figs/tree.png"), plot=otu_tree, width=8, height = 20)
```


