---
title: "rbcL community comparison"
output: html_notebook
---

```{r libs, include=F}
library(tidyverse)
library(here)
library(GUniFrac)
library(vegan)
library(ggrepel)
# library(vegalite)
# library(fossil)
library(picante)

# ggplot themes
library(ggthemes)
library(ggdark)
```

```{r input, include=F}
rbcl_rel_abund <- read_csv(here("data/clean/rbcl_rel_abund_clean.csv"))
idtax <- read_csv(here("data/clean/rbcl_taxonomy_clean.csv"))
field <- read_csv(here("data/clean/field_clean.csv"))

rbcl_asv_id_key <- read_csv(here("intermediate_output/rbcl_asv_key.csv"))
rbcl_asv_otu_key <- read_csv(here("intermediate_output/asv_otu_key.csv"))
alias <- read_csv(here("data/sample_aliases.csv"))
rbcl.tree <- read.tree(here("intermediate_output/tree/rbcl/asv613.fasta.treefile"))
```

```{r unifrac, include=F}
ra_wide <- rbcl_rel_abund %>% 
  select(asv_id, sample_id, rel_abund) %>% 
  pivot_wider(names_from = asv_id, values_from = rel_abund, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("sample_id")

# root tree with Trebouxiophyceae
root <- idtax %>% 
  filter(class=="Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(asv_id)

rbcl.tree <- rbcl.tree %>% root(outgroup = root, resolve.root=T)

# run weighted UniFrac
set.seed(123)
unifracs <- ra_wide %>% 
  GUniFrac::GUniFrac(rbcl.tree, alpha=1)
unifrac <- unifracs$unifracs[, , "d_1"]


# run NMDS ordination on results 
nmds <- metaMDS(unifrac, autotransform = F)
```


```{r unifrac_plot, echo=F}
# plot
unifrac_plot_data <- nmds %>% 
  scores() %>% 
  as_tibble(rownames = "sample_id") %>% 
  left_join(field, by="sample_id") %>% 
  left_join(alias, by="sample_id")

# plot unifrac, colored by elevation
uf_plot <- ggplot(unifrac_plot_data, aes(x=NMDS2, y=NMDS1)) +
  geom_point(aes(color=elev_m), size=4) +
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  labs(color="Elevation (m)") +
  geom_text_repel(aes(label=alias))
uf_plot
```



```{r setup_rel_abund_barplot, fig.width=10, fig.height=7, include=F}
plot_data <- rbcl_rel_abund %>% 
  left_join(rbcl_asv_otu_key, by="asv_id") %>% 
  group_by(sample_id, otu_name) %>% 
  summarise(rel_abund = sum(rel_abund)) %>%   
  left_join(field, by="sample_id") %>%
  left_join(alias, by="sample_id") %>% 
  ungroup() %>% 
  mutate(rel_abund = if_else(rel_abund <0.01, 0.01, rel_abund)) # artificially boost rel abund <1% to 1% so its visible on plot

# check rel abund sums = 1
plot_data %>%
  group_by(alias) %>%
  summarise(sum(rel_abund))
```


```{r plot_barplot, fig.width=10, fig.height=7, echo=F}
cust_theme <- theme(legend.position = "none",
                    panel.grid.major.x = element_blank(), # remove vertical lines in background
                    panel.grid.minor.x = element_blank(),
                    axis.line=element_blank(), # remove x axis
                    axis.title.x=element_blank(), 
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    panel.spacing = unit(-0.01, "lines"), # remove space between facet panels
                    strip.text.x = element_text(angle = 90, hjust=1.0, vjust = 0.8), # rotate facet labels 90
                    strip.background = element_blank()) # remove grey boxes around facet labs

# order OTUs
otu_ord <- c("Sanguina A","Chloromonas I","Chloromonas G","Chloromonas muramotoi E","Chloromonas krienitzii D","Chloromonas spp. C","Chlainomonas B","Chlainomonas F","Algae H", "Trebouxiophyceae J")

plot <- ggplot(plot_data, aes(x=alias %>% fct_reorder(elev_m), y=rel_abund)) +
  geom_bar(aes(fill=otu_name), stat="identity") + # classic stacked relative abundance plot
  coord_flip() +
  facet_grid(~otu_name %>% fct_relevel(otu_ord), switch = "x") +
  labs( x="Sample ID (from lowest to highest elevation)") +
  cust_theme
plot
# add sparkline on side for elevation
```

```{r faiths, echo=F}
re_wide_mat <- ra_wide %>% as.matrix()

faiths <- re_wide_mat %>% 
  picante::pd(rbcl.tree) %>% 
  # attach elevation data
  rownames_to_column("sample_id") %>% 
  left_join(field, by="sample_id") %>% 
  left_join(alias, by="sample_id")

faith_date_plot <- ggplot(faiths, aes(y=PD, x=date))+
  geom_point(aes(color=elev_m), size=4) +
  geom_smooth(method = "lm")+
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  geom_text_repel(aes(label=alias))+
  labs(title="Phylogenetic diversity of snow algae rbcL throughout the summer of 2018",y="Faith's phylogenetic diversity (PD)",x="Date", color="Elevation (m)")
faith_date_plot
```


```{r lm_faiths_pd, include=FALSE}
lm_out <- lm(PD ~ date + elev_m, data = faiths)
summary(lm_out)

# just elev
lm_out1 <- lm(log(PD) ~ elev_m, data = faiths) %>% # log transform to normalize
  broom::tidy()

elev_slope <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(estimate)
elev_p <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(p.value)

# just date
lm_out2 <- lm(PD ~ date, data = faiths) %>% 
  broom::tidy()

date_slope <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(estimate)
date_p <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(p.value)

# check resids
hist(resid(lm_out1))

# calculate co-linearity
car::vif(lm_out)
```


```{r output, include=F}
# ggsave
```

