---
title: "rbcL community comparison"
output: html_notebook
---

```{r, echo=FALSE}
knitr::opts_chunk$set(fig.path='figs/', echo=FALSE, warning=FALSE, message=FALSE)
```

```{r libs, include=F}
library(tidyverse)
library(here)
library(GUniFrac)
library(vegan)
library(ggrepel)
# library(vegalite)
# library(fossil)
library(picante)
library(gridExtra)

# ggplot themes
library(ggthemes)
library(ggdark)
```

```{r input, include=F}
rel_abund <- read_csv(here("data/03_filtered/rbcl/filtered_cleaned_rbcl_rel_abund.csv")) # not looking at the depth samples, controls, etc
tax <- read_csv(here("data/02_tidied/tidied.rbcl_taxonomy.csv"))
field <- read_csv(here("data/02_tidied/tidied_field.csv"))

asv_seqs <- read_csv(here("data/asv_seq_data/rbcl_asv_sequence_key.csv"))
otus <- read_csv(here("output/asv_otu_key.csv"))

rbcl.tree <- read.tree(here("output/tree/rbcl/all_asvs/rbcl_aligns_all.fasta.treefile")) ### this tree should only contain the asvs we found in the "main" sites, not looking at controls, etc


# the cell counts and 18S rel abund data
cellct_heatmap <- readRDS(here("output/cellct_w_seq_heatmap.Rda"))
s18_heatmap <- readRDS(here("output/s18_heatmap.Rda"))
```

```{r unifrac, include=F}

ra_wide <- rel_abund %>% 
  select(sample_id, asv_id, rel_abund) %>% 
  arrange(asv_id) %>% 
  pivot_wider(names_from = asv_id, values_from = rel_abund, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("sample_id")

# root tree with Trebouxiophyceae
root <- tax %>% 
  filter(class=="Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(asv_id)

rbcl.tree <- rbcl.tree %>% root(outgroup = root, resolve.root=T)


# check for differences between asvs in tree and rel abund table (all names must appear in tree)
asvs_rel_abund <- colnames(ra_wide)
asvs_tree <- rbcl.tree$tip.label %>% sort()

length(asvs_rel_abund)
length(asvs_tree)

setdiff(asvs_rel_abund, asvs_tree) # output elts unique to asvs_rel_abund

# run weighted UniFrac
set.seed(123)
unifracs <- ra_wide %>% 
  GUniFrac::GUniFrac(rbcl.tree, alpha=1) 
unifrac <- unifracs$unifracs[, , "d_1"]


# run NMDS ordination on results 
set.seed(123)
nmds <- metaMDS(unifrac, autotransform = F)
```


```{r unifrac_plot, echo=F, fig.width=5, fig.height=4}
# plot
unifrac_plot_data <- nmds %>% 
  scores() %>% 
  as_tibble(rownames = "sample_id") %>% 
  left_join(field, by="sample_id")

# plot unifrac, colored by elevation
uf_plot <- ggplot(unifrac_plot_data, aes(x=NMDS2, y=NMDS1)) +
  geom_point(aes(color=elev_m), size=4) +
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  labs(color="Elevation (m)", tag="A") +
  geom_text_repel(aes(label=alias), size=3, segment.size = 0.3) +
  theme_classic()
uf_plot
```



```{r setup_rel_abund_barplot, include=F}
otu_rel_abund <- rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  group_by(sample_id, otu_name) %>% 
  summarise(rel_abund = sum(rel_abund)) %>%   
  left_join(field, by="sample_id") %>%
  ungroup()

# otu_rel_abund_round_to_1_percent <- otu_rel_abund %>% 
#   mutate(rel_abund = if_else(rel_abund <0.01, 0.01, rel_abund)) # artificially boost rel abund <1% to 1% so its visible on plot
# 
# # check rel abund sums ~= 1 (will be slightly higher, due to rounding)
# otu_rel_abund_round_to_1_percent %>%
#   group_by(alias) %>%
#   summarise(sum(rel_abund))
```


```{r plot_barplot, echo=F, fig.width=2.9, fig.height=5}
otu_ord <- c("Chlainomonas A","Sanguina B", "Other Chloromonas","Chloromonas krienitzii","Chloromonas muramotoi C","Chloromonas D", "Unknown E","Chloromonas F", "Trebouxiophyceae G")

my_theme <- theme(axis.text.x = element_text(angle = 45, hjust = 1))

otu_rel_abund %>% distinct(sample_id) %>% nrow()

rbcl_heatmap <- ggplot(otu_rel_abund, aes(x=otu_name %>% fct_relevel(otu_ord), y=alias %>% fct_reorder(elev_m))) +
  geom_tile(aes(alpha=rel_abund, fill=otu_name %>% fct_relevel(otu_ord))) +
  labs(x="rbcL OTU", y= "Elevation", fill = "rbcL OTU", alpha="Relative abundance") +
  scale_alpha(range = c(0.01, 1)) +
  my_theme +
  guides(alpha="none", fill="none")
rbcl_heatmap
```


```{r colors, include=F}
# get colors, so they match across plots
g <- ggplot_build(rbcl_heatmap)

colors <- tibble(otu_name = otu_rel_abund$otu_name,
                 color = g$data[[1]]$fill) %>% 
  distinct(otu_name, color) %>% 
  deframe()

length(colors)
colors

```

Now add in the 18s and cellcount data side by side the rbcL OTU relative abundance data
- set colors same
- add in missing cellcount data
- plot left to right (entire width of page): rbcl, cellct, 18s, with rbcl taking up most width
- 
```{r 18s_rel_abund, fig.width=1.7, fig.height=5}
s18_heat <- s18_heatmap +
  scale_fill_manual(values=c("#D39200", "#93AA00", "#FF61C3","#808080"))
s18_heat
```

```{r cellct, fig.width=2.6, fig.height=5}
cellct_heat <- cellct_heatmap +
  scale_fill_manual(values = c("#F8766D","#D39200","#00BA38","#93AA00","#93AA00","#808080","#808080","#808080"))
cellct_heat
```

```{r elev_strip, fig.width=1.5, fig.height=5}
my_theme <- theme_classic() +
  theme(axis.line=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        axis.title.y = element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())
  
elev_strip_tbl <- otu_rel_abund %>% 
  distinct(alias, elev_m) %>% 
  arrange(-elev_m) %>% 
  add_column(x="column_1") %>% 
  mutate(elev_m = elev_m %>% round(digits=-1))
nrow(elev_strip_tbl)

elev_strip <- ggplot(elev_strip_tbl, aes(x=x, y=alias %>% fct_reorder(elev_m))) +
  geom_tile(aes(fill=elev_m)) +
  scale_fill_gradientn(colours = terrain.colors(8)[1:7]) +
  labs(x="Elevation (m)") +
  my_theme 
elev_strip
```


```{r elev_strip2, fig.width=0.5, fig.height=5}
elev_strip2 <- elev_strip +
  guides(fill="none") +
  geom_text(aes(label=elev_m), size=3)
elev_strip2
```

```{r snow_color_strip, fig.width=0.5, fig.height=5}
color_strip_tbl <- otu_rel_abund %>% 
  arrange(-elev_m) %>% 
  distinct(alias, color_snow, elev_m) %>% 
  add_column(x="column_1")

color_strip_tbl %>% distinct(color_snow)

color_strip <- ggplot(color_strip_tbl, aes(x=x, y=alias %>% fct_reorder(elev_m))) +
  geom_tile(aes(fill=color_snow)) +
  my_theme +
  guides(fill="none") +
  scale_fill_manual(values=c(pink="red",other="grey",green="green",orange="orange",rusty="purple"))
color_strip
```



Faith's Phylogenetic Diversity, sum of branch lengths in a sample's phylogenetic tree (does not take relative abundance into account)
```{r faiths, echo=F}
ra_wide_mat <- ra_wide %>% as.matrix()

faiths <- ra_wide_mat %>% 
  picante::pd(rbcl.tree) %>% 
  # attach elevation data
  rownames_to_column("sample_id") %>% 
  left_join(field, by="sample_id")

faith_date_plot <- ggplot(faiths, aes(y=PD, x=date))+
  geom_point(aes(color=elev_m), size=4) +
  geom_smooth(method = "lm")+
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  geom_text_repel(aes(label=alias))+
  labs(title="Phylogenetic diversity of snow algae rbcL throughout the summer of 2018",y="Faith's phylogenetic diversity (PD)",x="Date", color="Elevation (m)")
faith_date_plot
```

```{r, echo=F}
faith_elev_plot <- ggplot(faiths, aes(y=PD, x=elev_m))+
  geom_point() +
  geom_smooth(method = "lm")+
  geom_text_repel(aes(label=alias))+
  labs(y="Faith's phylogenetic diversity (PD)",x="Elevation (m)", tag="B")
faith_elev_plot
```

Shannon and Simpson diversity indices (these take relative abundance into account, but require OTUs)
```{r, include=F}
otu_rel_abund_mat <- otu_rel_abund %>% 
  select(sample_id, otu_name, rel_abund) %>% 
  pivot_wider(names_from = otu_name, values_from = rel_abund, values_fill = list(rel_abund=0)) %>% 
  column_to_rownames("sample_id") %>% 
  as.matrix()

# use ra_wide_mat for asv based diversity
shannon <- otu_rel_abund_mat %>% 
  diversity(index = "shannon") %>% 
  enframe() %>% 
  dplyr::rename(sample_id = name, shannon = value)
simpson <- otu_rel_abund_mat %>% 
  diversity(index = "simpson") %>% 
  enframe() %>% 
  dplyr::rename(sample_id = name, simpson = value)
otu_diversity <- shannon %>% 
  left_join(simpson, by="sample_id") %>% 
  left_join(field, by="sample_id")

# plot diversity of samples
shannon <- ggplot(otu_diversity, aes(x=elev_m, y=shannon))+
  geom_point() +
  geom_text_repel(aes(label=alias)) +
  geom_smooth(method = "lm")+
  labs(y="Shannon diversity", tag="A")+
  theme(axis.title.x = element_blank())
shannon
```

```{r diversity_plots, include=F}
diversity_plots <- gridExtra::grid.arrange(shannon, faith_elev_plot, nrow=2)

```



Chao
```{r}
# otu_rel_abund_mat %>% 
#   mutate
#   # head(1) %>% 
#   as.matrix() %>% 
#   fossil::chao1()
```


```{r lm_faiths_pd, include=FALSE}
lm_out <- lm(PD ~ date + elev_m, data = faiths)
summary(lm_out)

# just elev
lm_out1 <- lm(log(PD) ~ elev_m, data = faiths) %>% # log transform to normalize
  broom::tidy()

elev_slope <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(estimate)
elev_p <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(p.value)

# just date
lm_out2 <- lm(PD ~ date, data = faiths) %>% 
  broom::tidy()

date_slope <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(estimate)
date_p <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(p.value)

# check resids
hist(resid(lm_out1))

# calculate co-linearity
car::vif(lm_out)
```


```{r output, include=F}

ggsave(here("manuscript/figs/main/community/rbcl.pdf"),plot=rbcl_heatmap, width = 2.9, height = 5)
ggsave(here("manuscript/figs/main/community/18s.pdf"),plot=s18_heat, width = 1.7, height = 5)
ggsave(here("manuscript/figs/main/community/cellct.pdf"),plot=cellct_heat, width = 2.6, height = 5)
ggsave(here("manuscript/figs/main/community/elev.pdf"),plot=elev_strip, width = 1.5, height = 5)
ggsave(here("manuscript/figs/main/community/elev2.pdf"),plot=elev_strip2, width = 0.5, height = 5)
ggsave(here("manuscript/figs/main/community/snow_color.pdf"),plot=color_strip, width = 0.5, height = 5)


ggsave(filename = here("manuscript/figs/main/uf_plot.png"), plot=uf_plot, width=5, height = 4)
ggsave(filename = here("manuscript/figs/diversity.png"), plot = diversity_plots, height = 10, width = 7)
```

