---
title: "rbcL community comparison"
output: html_notebook
---

```{r, echo=FALSE}
knitr::opts_chunk$set(fig.path='figs/', echo=FALSE, warning=FALSE, message=FALSE)
```

```{r libs, include=F}
library(tidyverse)
library(here)
library(GUniFrac)
library(vegan)
library(ggrepel)
# library(vegalite)
# library(fossil)
library(picante)
library(gridExtra)

# ggplot themes
library(ggthemes)
library(ggdark)
```

```{r input, include=F}
rel_abund <- read_csv(here("data/03_filtered/rbcl/filtered_cleaned_rbcl_rel_abund.csv")) # not looking at the depth samples, controls, etc
tax <- read_csv(here("data/02_tidied/tidied.rbcl_taxonomy.csv"))
field <- read_csv(here("data/02_tidied/tidied_field.csv"))

asv_seqs <- read_csv(here("data/asv_seq_data/rbcl_asv_sequence_key.csv"))
otus <- read_csv(here("output/asv_otu_key.csv"))

rbcl.tree <- read.tree(here("output/tree/rbcl/all_asvs/rbcl_aligns_all.fasta.treefile")) ### this tree should only contain the asvs we found in the "main" sites, not looking at controls, etc
```

```{r unifrac, include=F}

ra_wide <- rel_abund %>% 
  select(sample_id, asv_id, rel_abund) %>% 
  arrange(asv_id) %>% 
  pivot_wider(names_from = asv_id, values_from = rel_abund, values_fill = list(rel_abund = 0)) %>% 
  column_to_rownames("sample_id")

# root tree with Trebouxiophyceae
root <- tax %>% 
  filter(class=="Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(asv_id)

rbcl.tree <- rbcl.tree %>% root(outgroup = root, resolve.root=T)


# check for differences between asvs in tree and rel abund table (all names must appear in tree)
asvs_rel_abund <- colnames(ra_wide)
asvs_tree <- rbcl.tree$tip.label %>% sort()

length(asvs_rel_abund)
length(asvs_tree)

setdiff(asvs_rel_abund, asvs_tree) # output elts unique to asvs_rel_abund

# run weighted UniFrac
set.seed(123)
unifracs <- ra_wide %>% 
  GUniFrac::GUniFrac(rbcl.tree, alpha=1) 
unifrac <- unifracs$unifracs[, , "d_1"]


# run NMDS ordination on results 
set.seed(123)
nmds <- metaMDS(unifrac, autotransform = F)
```


```{r unifrac_plot, echo=F}
# plot
unifrac_plot_data <- nmds %>% 
  scores() %>% 
  as_tibble(rownames = "sample_id") %>% 
  left_join(field, by="sample_id")

# plot unifrac, colored by elevation
uf_plot <- ggplot(unifrac_plot_data, aes(x=NMDS2, y=NMDS1)) +
  geom_point(aes(color=elev_m), size=4) +
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  labs(color="Elevation (m)", tag="A") +
  geom_text_repel(aes(label=alias))
uf_plot
```



```{r setup_rel_abund_barplot, include=F}
otu_rel_abund <- rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  group_by(sample_id, otu_name) %>% 
  summarise(rel_abund = sum(rel_abund)) %>%   
  left_join(field, by="sample_id") %>%
  ungroup()

otu_rel_abund_round_to_1_percent <- otu_rel_abund %>% 
  mutate(rel_abund = if_else(rel_abund <0.01, 0.01, rel_abund)) # artificially boost rel abund <1% to 1% so its visible on plot

# check rel abund sums ~= 1 (will be slightly higher, due to rounding)
otu_rel_abund_round_to_1_percent %>%
  group_by(alias) %>%
  summarise(sum(rel_abund))
```


```{r plot_barplot, echo=F}
cust_theme <- theme(legend.position = "none",
                    panel.grid.major.x = element_blank(), # remove vertical lines in background
                    panel.grid.minor.x = element_blank(),
                    axis.line=element_blank(), # remove x axis
                    axis.title.x=element_blank(), 
                    axis.text.x=element_blank(),
                    axis.ticks.x=element_blank(),
                    panel.spacing = unit(-0.01, "lines"), # remove space between facet panels
                    strip.text.x = element_text(angle = 90, hjust=1.0, vjust = 0.8), # rotate facet labels 90
                    strip.background = element_blank()) # remove grey boxes around facet labs

otus %>% distinct(otu_name)


otu_ord <- c("Chlainomonas A","Sanguina B", "Other Chloromonas","Chloromonas krienitzii","Chloromonas muramotoi C","Chloromonas D", "Algae E","Chloromonas F", "Trebouxiophyceae G")

barplot <- ggplot(otu_rel_abund_round_to_1_percent, aes(x=alias %>% fct_reorder(elev_m), y=rel_abund)) +
  geom_bar(aes(fill=otu_name %>% fct_relevel(otu_ord)), stat="identity") + # classic stacked relative abundance plot
  coord_flip() +
  facet_grid(~otu_name %>% fct_relevel(otu_ord), switch = "x") +
  labs( x="Sample ID", tag="B") +
  cust_theme
barplot

```

```{r combine, fig.width=7, fig.height=10}
combined_fig_3 <- gridExtra::grid.arrange(uf_plot, barplot, nrow=2)
```


Faith's Phylogenetic Diversity, sum of branch lengths in a sample's phylogenetic tree (does not take relative abundance into account)
```{r faiths, echo=F}
ra_wide_mat <- ra_wide %>% as.matrix()

faiths <- ra_wide_mat %>% 
  picante::pd(rbcl.tree) %>% 
  # attach elevation data
  rownames_to_column("sample_id") %>% 
  left_join(field, by="sample_id")

faith_date_plot <- ggplot(faiths, aes(y=PD, x=date))+
  geom_point(aes(color=elev_m), size=4) +
  geom_smooth(method = "lm")+
  scale_color_gradientn(colours = terrain.colors(8)[1:7]) +
  geom_text_repel(aes(label=alias))+
  labs(title="Phylogenetic diversity of snow algae rbcL throughout the summer of 2018",y="Faith's phylogenetic diversity (PD)",x="Date", color="Elevation (m)")
faith_date_plot
```

```{r, echo=F}
faith_elev_plot <- ggplot(faiths, aes(y=PD, x=elev_m))+
  geom_point() +
  geom_smooth(method = "lm")+
  geom_text_repel(aes(label=alias))+
  labs(title="Phylogenetic diversity of snow algae rbcL vs elevation",y="Faith's phylogenetic diversity (PD)",x="Elevation (m)")
faith_elev_plot
```

Shannon and Simpson diversity indices (these take relative abundance into account, but require OTUs)
```{r, include=F}
otu_rel_abund_mat <- otu_rel_abund %>% 
  select(sample_id, otu_name, rel_abund) %>% 
  pivot_wider(names_from = otu_name, values_from = rel_abund, values_fill = list(rel_abund=0)) %>% 
  column_to_rownames("sample_id") %>% 
  as.matrix()

# use ra_wide_mat for asv based diversity
shannon <- otu_rel_abund_mat %>% 
  diversity(index = "shannon") %>% 
  enframe() %>% 
  dplyr::rename(sample_id = name, shannon = value)
simpson <- otu_rel_abund_mat %>% 
  diversity(index = "simpson") %>% 
  enframe() %>% 
  dplyr::rename(sample_id = name, simpson = value)
otu_diversity <- shannon %>% 
  left_join(simpson, by="sample_id") %>% 
  left_join(field, by="sample_id")

# plot diversity of samples
ggplot(otu_diversity, aes(x=elev_m, y=shannon))+
  geom_point() +
  geom_text_repel(aes(label=alias)) +
  geom_smooth(method = "lm")
```


Chao
```{r}
# otu_rel_abund_mat %>% 
#   mutate
#   # head(1) %>% 
#   as.matrix() %>% 
#   fossil::chao1()
```


```{r lm_faiths_pd, include=FALSE}
lm_out <- lm(PD ~ date + elev_m, data = faiths)
summary(lm_out)

# just elev
lm_out1 <- lm(log(PD) ~ elev_m, data = faiths) %>% # log transform to normalize
  broom::tidy()

elev_slope <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(estimate)
elev_p <- lm_out1 %>% 
  filter(term == "elev_m") %>% 
  pull(p.value)

# just date
lm_out2 <- lm(PD ~ date, data = faiths) %>% 
  broom::tidy()

date_slope <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(estimate)
date_p <- lm_out2 %>% 
  filter(term == "date") %>% 
  pull(p.value)

# check resids
hist(resid(lm_out1))

# calculate co-linearity
car::vif(lm_out)
```


```{r output, include=F}
ggsave(filename = here("manuscript/figs/3_community/combined_fig_3.png"), plot=combined_fig_3, height=10, width=7)
ggsave(filename = here("manuscript/figs/supplementary/diversity.png"), plot = faith_date_plot, height = 5, width = 7)
```

