---
title: "rbcL ASV clustering"
output: html_notebook
---


```{r libs, include=F}
library(tidyverse)
library(here)
library(DECIPHER)
library(ggforce)
library(Rtsne)
library(ape)
library(ggtree)
library(treeio)
library(dbscan)
library(ggrepel)
library(RColorBrewer)
library(scales)

library(ggpubr)
library(ggthemes)
library(ggdark)

conflicted::conflict_scout()
conflicted::conflict_prefer("filter","dplyr")
```

```{r read, include=F}
rel_abund <- read_csv(here("data/03_filtered/rbcl/filtered_cleaned_rbcl_rel_abund.csv"))
taxonomy <- read_csv(here("data/02_tidied/tidied.rbcl_taxonomy.csv"))

aligns <- readDNAStringSet(here("output/alignments/rbcl_aligns_in_main.fasta"))
tree <- read.tree(file=here("output/tree/rbcl/asv613.fasta.treefile")) # this is the tree with only the ASVs that appear in the filtered rel abund dataset (not including protocol stuff, etc)

# actual seq data
key <- read_csv(here("data/asv_seq_data/rbcl_asv_sequence_key.csv"))
```

```{r asv_distance_matrix, include=F}
BrowseSeqs(aligns, highlight = T)

dists <-  aligns %>% 
  DECIPHER::DistanceMatrix(type="matrix", correction = "JC")
```

```{r tsne_1, echo=F}
perplexity <- 30

set.seed(123)
tsne_full_output <- dists %>% 
  Rtsne(pca=F, perplexity = perplexity)

tsne_pts <- tsne_full_output$Y %>% 
  as_tibble()

tsne_pts_taxonomy <- tsne_pts %>% 
  add_column(asv_id = colnames(dists)) %>% # label each point by it's ASV ID
  left_join(taxonomy, by="asv_id")

taxa.order <- c("Chloromonas","Chloromonas_krienitzii","Chloromonas_muramotoi","Chloromonas_hohamii","Chloromonas_Group_B","other Trebouxiophyceae","Raphidonema_nivale","Stichococcus","Chlainomonas","Chlainomonas_rubra","Sanguina","Sanguina_nivaloides","Sanguina_aurantia")

ggplot(tsne_pts_taxonomy, aes(x=V1, y=V2)) +
  # plot points not assigned to genus or species
  geom_point(data = filter(tsne_pts_taxonomy, is.na(plot_taxa)), 
             color = "grey", alpha = 0.8) +
  # color points that were assigned to genus or species
  geom_point(data = filter(tsne_pts_taxonomy, !is.na(plot_taxa)), 
             aes(color = plot_taxa %>% fct_relevel(taxa.order)), 
             alpha = 0.8)
```


```{r otu-cluster, echo=F}
eps <- 4

tsne_mat <- tsne_pts_taxonomy %>% 
  select(1,2) %>% 
  as.matrix()

set.seed(123)

dbscan_out <- dbscan::dbscan(tsne_mat, eps=eps) # eps=4 gives best results

tsne_taxa_otu <- tsne_pts_taxonomy %>% 
  add_column(otu = as.character(dbscan_out$cluster)) # tsne is in order of asv_id

ggplot(tsne_taxa_otu, aes(x=V1, y=V2)) +
  geom_point(aes(color = otu)) +
  labs(title="dbscan clustering")
```

Manually edit the clustering to show *Chloromonas krienitzii* on relative abundance barplot

```{r edit-otus, echo=F}
tsne_edited_otus <- tsne_taxa_otu %>%
  mutate(otu = case_when(plot_taxa == "Chloromonas_krienitzii" ~ "9",
                         TRUE ~ otu))

otus <- tsne_edited_otus %>% 
  select(asv_id, otu)

# get names for each otu
rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  left_join(taxonomy, by="asv_id") %>% 
  distinct(otu, plot_taxa) %>% 
  arrange(otu)

# get rel abund of each otu
rel_abund %>% 
  left_join(otus, by="asv_id") %>% 
  group_by(otu) %>% 
  summarise(rel_abund = sum(rel_abund)) %>% 
  arrange(-rel_abund) %>% 
  select(-rel_abund) %>% 
  add_column(letter = LETTERS[1:9])

# name each OTU, lettered in order of rel abund
otu_names <- tibble(otu = as.character(1:9), 
                    otu_name = c("Other Chloromonas",
                                 "Sanguina B",
                                 "Chlainomonas A",
                                 "Chloromonas muramotoi C",
                                 "Chloromonas D",
                                 "Chloromonas F",
                                 "Unknown E",
                                 "Trebouxiophyceae G",
                                 "Chloromonas krienitzii"))

otus_asv_name <- otus %>% 
  left_join(otu_names, by="otu") %>% 
  select(asv_id, otu_name)

tsne_added_names <- tsne_edited_otus %>% 
  left_join(otus_asv_name, by="asv_id")
```


```{r include_rel_abund, echo=F, fig.width=5, fig.height=4}
total_rel_abund <- rel_abund %>% 
  group_by(asv_id) %>% 
  summarise(global_rel_abund = sum(rel_abund))

tsne_rel_abund <- tsne_added_names %>% 
  left_join(total_rel_abund, by="asv_id") %>% 
  # remove C kri ellipse
  mutate(otu_name = if_else(otu_name == "Chloromonas krienitzii", "Other Chloromonas", otu_name))


final_tsne <- ggplot(tsne_rel_abund, aes(x=V1, y=V2)) +
  ggforce::geom_mark_ellipse(aes(fill=otu_name, label=otu_name), 
                             show.legend = FALSE, 
                             alpha=0, 
                             linetype="dashed", 
                             label.margin = margin(0,0,0,0,"mm"),
                             label.buffer = unit(1, "mm"),
                             label.fontface = "plain",
                             label.fontsize = 8,
                             con.type = "none",
                             expand = unit(2, "mm")) + # expansion of ellipse around points
  geom_point(aes(color = plot_taxa %>% fct_relevel(taxa.order), size = global_rel_abund, alpha=0.5))+
  labs(color="ASV taxonomic assignment", tag="B", title="t-SNE") +
  ggthemes::theme_few() +
  scale_size(range = c(2, 12), guide="none") 

tsne_no_legend <- final_tsne +
    theme(legend.position = "none")
tsne_no_legend
```

```{r, fig.width=6,fig.height=8, include=F}
taxa.order
display.brewer.all()
scales::show_col(brewer_pal(palette = "Set1")(5))
```

13 taxa, 9 OTUs
Taxa colors: (* if representative of OTU)
- Chloromonas (5 species, 5 OTU)
  - unspecified*
  - krienitzii*
  - muramotoi*
  - hohamii (1) --- D
  - Group B (1) --- F
- Chlainomonas (2 sp, 1 otu)
  - unspecified*
  - rubra
- Sanguina (3 sp, 1 otu)
  - unspecified*
  - nivaloides
  - aurentia
- Trebouxiophyceae (3 sp, 1 otu)
  - unspecified*
  - Raphidonema
  - Stichococcus
  
4 diverging colors for the different genera, use shades of colors for species. The darkest color is "unspecified", use this color for the OTUs. 

Strategy: use green and orange for Chloromonas, Red for Sanguina, Purple for Chlainomonas, blye for trebouxiophyceae


```{r, fig.width=2, fig.height=2}
chloromonas_color <- "#4daf4a" # green
krienitzii_color <- "#ff7f00" # orange
muramotoi_color <- "#d7d42e" # yellow
# pick shades using https://www.color-hex.com/
hohamii_and_D_color <- "#265725" # dark green
group_B_and_F_color <- "#b25800" # brown
chloromonas_colors <- c(chloromonas_color, krienitzii_color, muramotoi_color, hohamii_and_D_color,group_B_and_F_color)
scales::show_col(chloromonas_colors)
```


```{r,fig.width=2, fig.height=2}
trebouxiophyceae_color <- "#377eb8" # blu
raphidonema_color <- "#9bbedb" # light
stichococcus_color <- "#1b3f5c" # dark
trebouxiophyceae_colors <- c(trebouxiophyceae_color,stichococcus_color,raphidonema_color)
scales::show_col(trebouxiophyceae_colors)
```


```{r,fig.width=2, fig.height=2}
chlainomonas_color <- "#984ea3" # purp
rubra_color <- "#c194c7" #lilac
chlainomonas_colors <- c(chlainomonas_color,rubra_color)
scales::show_col(chlainomonas_colors)
```


```{r,fig.width=2, fig.height=2}
# changed from colorbrewer red = too agressive
sanguina_color <- "#ff534a"
nivaloides_color <- "#ff8680"
aurentia_color <- "#b23a33"
sanguina_colors <- c(sanguina_color,nivaloides_color,aurentia_color)
scales::show_col(sanguina_colors)
```


```{r,fig.width=2, fig.height=2}
na_color <- "grey70"

taxa_colors <- c(chloromonas_colors,trebouxiophyceae_colors,chlainomonas_colors, sanguina_colors,na_color)
names(taxa_colors) <- taxa.order
scales::show_col(taxa_colors)
```


```{r,fig.width=2, fig.height=2}
otu_colors <- c(chlainomonas_color,sanguina_color,chloromonas_colors[1:2],"#173416",chloromonas_colors[4:5], trebouxiophyceae_color,na_color)
# manually change muramotoi color to dark green so it shows up on the heatmap
names(otu_colors) <- c("Chlainomonas A","Sanguina B","Other Chloromonas","Chloromonas krienitzii","Chloromonas muramotoi C","Chloromonas D","Chloromonas F","Trebouxiophyceae G","Unknown E")
scales::show_col(otu_colors)
```

Change colors in plot
```{r}
tsne_no_legend_nice_colors <- tsne_no_legend +
  scale_color_manual(values=taxa_colors, na.value=na_color)
tsne_no_legend_nice_colors
```



```{r, fig.height=4, fig.width=2.2}
p <- final_tsne +
  scale_color_manual(values=taxa_colors, na.value=na_color) +
  guides(colour = guide_legend(override.aes = list(alpha = 1, size=3)), alpha="none") 
legend <- p %>% 
  get_legend() %>% 
  as_ggplot()
legend
```


```{r asv_mds, echo=F, fig.width=5, fig.height=4}
set.seed(123)
mds_coords <- dists %>% 
  cmdscale() %>% # ok to use euclidian dist, no null values when comparing relatedness
  as_tibble() %>% 
  add_column(asv_id = colnames(dists), .before=1)

mds_asv_plot <- mds_coords %>% 
  left_join(taxonomy, by="asv_id") %>% 
  left_join(otus_asv_name, by="asv_id") %>% 
  left_join(total_rel_abund, by="asv_id") %>% 
  filter(!(asv_id %in% c("639", "534", "646"))) # remove Trebouxiophyceae outliers

mds <- ggplot(mds_asv_plot, aes(V1, V2)) + 
  ggforce::geom_mark_ellipse(data = mds_asv_plot %>% 
                               filter(otu_name != "Trebouxiophyceae J"), # dont plot the ellipse around other treb
                             aes(fill = otu_name, label=otu_name),
                             show.legend = FALSE, 
                             alpha=0, 
                             linetype="dashed", 
                             label.margin = margin(0,0,0,0,"mm"),
                             label.buffer = unit(1, "mm"),
                             label.fontface = "plain",
                             label.fontsize = 8,
                             con.type = "none",
                             expand = unit(2, "mm")) +
  geom_point(aes(color = plot_taxa %>% fct_relevel(taxa.order), size = global_rel_abund, alpha=0.5))+
  labs(color="ASV taxonomic assignment", tag="A",title="MDS") +
  ggthemes::theme_few() +
  scale_size(range = c(2, 12), guide="none")+
  guides(colour = guide_legend(override.aes = list(alpha = 1, size=3)), alpha="none") +
  theme(legend.position = "none") +
  scale_color_manual(values=taxa_colors, na.value=na_color)
  
mds
```

```{r, fig.width=4, fig.height=6}
asv_arranged <- gridExtra::grid.arrange(mds, tsne_no_legend_nice_colors, nrow=2)
```



# Phylogenetic tree


Run IQtree
```{r tree, fig.width=7, fig.height=10}
# root tree with Trebouxiophyceae
root <- taxonomy %>% 
  filter(class=="Trebouxiophyceae") %>% 
  head(1) %>% 
  pull(asv_id)

rbcl.tree <- rbcl.tree %>% root(outgroup = root, resolve.root=T)



rbcl_ggtree <- ggtree(rbcl.tree)

# parse bootstrap values
rbcl_ggtree_data <- rbcl_ggtree$data %>% 
  filter(isTip == FALSE) %>% 
  separate(label, c("boot", "bayes"), sep = "/") %>% #split label into boots and bayes vals
  filter(boot > 80) %>%
  mutate(boot = round(as.numeric(boot), digits = 0),
         insignificant_edge = if_else(boot > 80, "black", "red"))# round boot to nearest integer

# Plot trees

# tips colored by taxonomy
taxonomy_tree <- rbcl_ggtree %<+% taxonomy + 
  geom_tippoint(aes(color=plot_taxa, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

# clades annotated by OTU
otu_tree <- rbcl_ggtree %<+% otus_asv_name + 
  geom_tippoint(aes(color=otu_name, alpha=0.7), size=3) + # 
  geom_text2(data=rbcl_ggtree_data, aes(label=paste0(boot,";",bayes)),  vjust= 0.01, hjust = 1, size=3) + # add bootstrap vals
  theme(legend.position = c(0.2,0.7), 
        legend.title = element_blank(), 
        legend.key = element_blank()) 

taxonomy_tree
otu_tree

```

todo: 
- better color choices
- circular tree


# make Fastas for breanna and jordan
```{r setup_fastas}
rbcl_seqs_otus <- otus_asv_name %>% 
  left_join(key, by="asv_id")
  
chlain_seqs <- rbcl_seqs_otus %>% 
  filter(otu_name =="Chlainomonas A") %>% 
  select(-otu_name) %>% 
  deframe() %>% 
  DNAStringSet()
names(chlain_seqs) <- paste0("field_miseq_",names(chlain_seqs))

sang_seqs <- rbcl_seqs_otus %>% 
  filter(otu_name =="Sanguina B") %>% 
  select(-otu_name) %>% 
  deframe() %>% 
  DNAStringSet()
names(sang_seqs) <- paste0("field_miseq_",names(sang_seqs))

# also just the ones that matched the database
sang_taxa_seqs <- taxonomy %>% 
  filter(genus=="Sanguina") %>% 
  select(asv_id) %>% 
  left_join(key, by="asv_id") %>% 
  mutate(asv_id = paste0("field_miseq_",asv_id)) %>% 
  deframe() %>% 
  DNAStringSet()

chlain_taxa_seqs <- taxonomy %>% 
  filter(genus=="Chlainomonas") %>% 
  select(asv_id) %>% 
  left_join(key, by="asv_id") %>% 
  mutate(asv_id = paste0("field_miseq_",asv_id)) %>% 
  deframe() %>% 
  DNAStringSet()

```



```{r output}
write_csv(otus_asv_name, path = here("output/asv_otu_key.csv"))

# write the main ASV fig, w legend seperate
ggsave(filename = here("manuscript/figs/main/taxonomy/asvs.pdf"), plot=asv_arranged, height = 6, width = 4)
ggsave(filename = here("manuscript/figs/main/taxonomy/asv_legend.pdf"), plot=legend,  height = 4, width = 2.2)

# write out the trees
ggsave(filename = here("manuscript/figs/otu_tree.png"), plot=otu_tree, width=7, height = 10)
ggsave(filename = here("manuscript/figs/taxonomy_tree.png"), plot=taxonomy_tree, width=7, height = 10) 

# write out otu colors
otu_colors %>% 
  enframe() %>% 
  write_csv(here("output/etc/otu_colors.csv"))

# sanguina fasta for breanna
writeXStringSet(sang_seqs, here("output/taxa_filtered_fastas/sanguina_otu.fasta"))
writeXStringSet(sang_taxa_seqs, here("output/taxa_filtered_fastas/sanguina_idtaxa.fasta"))


# chlainomonas fasta for jordan
writeXStringSet(chlain_seqs, here("output/taxa_filtered_fastas/chlainomonas_otu.fasta"))
writeXStringSet(chlain_taxa_seqs, here("output/taxa_filtered_fastas/chlainomonas_idtaxa.fasta"))
```


